{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<script>\n",
       "code_show=true; \n",
       "function code_toggle() {\n",
       " if (code_show){\n",
       " $('div.input').hide();\n",
       " } else {\n",
       " $('div.input').show();\n",
       " }\n",
       " code_show = !code_show\n",
       "} \n",
       "$( document ).ready(code_toggle);\n",
       "</script>\n",
       "<b>The raw code for this IPython notebook is by default hidden for easier reading.To toggle on/off the raw code, click <a href=\"javascript:code_toggle()\">here</a></b>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from IPython.display import HTML\n",
    "\n",
    "HTML('''<script>\n",
    "code_show=true; \n",
    "function code_toggle() {\n",
    " if (code_show){\n",
    " $('div.input').hide();\n",
    " } else {\n",
    " $('div.input').show();\n",
    " }\n",
    " code_show = !code_show\n",
    "} \n",
    "$( document ).ready(code_toggle);\n",
    "</script>\n",
    "<b>The raw code for this IPython notebook is by default hidden for easier reading.To toggle on/off the raw code, click <a href=\"javascript:code_toggle()\">here</a></b>''')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Meta Analysis of the Datasets for the EpiÂ² pilot project \n",
    "\n",
    "---\n",
    "# lncRNA interaction with proteins\n",
    "\n",
    "---\n",
    "# PYTHON 3 / R Notebook\n",
    "### Adrien Leger / EMBL EBI\n",
    "### Starting date 18/08/2016"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "## Import general package and definition of specific functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>.container { width:75% !important; }</style>"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Standard lib imports\n",
    "from glob import glob\n",
    "from os import remove\n",
    "\n",
    "# Local lib import\n",
    "import pycl\n",
    "import pyBioPlot as pbl\n",
    "\n",
    "# Third party import\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import pylab as pl\n",
    "import seaborn as sns\n",
    "import scipy.stats as stats\n",
    "\n",
    "# Pyplot tweaking\n",
    "%matplotlib inline\n",
    "pl.rcParams['figure.figsize'] = 30, 10  # that's default image size for this interactive session\n",
    "\n",
    "# Larger display \n",
    "from IPython.core.display import display, HTML\n",
    "display(HTML(\"<style>.container { width:75% !important; }</style>\"))\n",
    "\n",
    "# Allow to use R directly\n",
    "%load_ext rpy2.ipython\n",
    "\n",
    "# Simplify warning reporting to lighten the notebook style \n",
    "import warnings\n",
    "warnings.formatwarning=(lambda message, *args: \"{}\\n\".format(message))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "## Hendrickson = fRIP with 24 chromatin related protein"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***Hendrickson D G, DR Kelley, D Tenen, B Bernstein, JL Rinn. Widespread RNA binding by chromatin-associated proteins.. Genome Biol 17, 28 (2016).***\n",
    "\n",
    "- Formaldehyde RNA immunoprecipitation of 24 RPB and chromatin related proteins \n",
    "- Really detaillled and extensive datasets to have transcriptome wide lncRNA interacting with chromatin.\n",
    "- Contain all the list of lncRNA gene significantly enriched in captured factions\n",
    "- Problem = K562 (myelogenous leukemia line) with highly rearranged genomes...\n",
    "    \n",
    "=> The data provided will be hard to analyse and cannnot simply be lifovered\n",
    "\n",
    "=> I would have to re-analign the fastq files at transcript level, compare bound and unbound fractions, call the peaks...\n",
    "\n",
    "I am not sure that this is worth to do it"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "## Li  / Starbase =  CLIP-Seq data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "***JH Li, S Liu, LL Zheng, J Wu, WJ Sun, ZL Wang, H Zhou, LH Qu, JH Yang. Discovery of Protein-lncRNA Interactions by Integrating Large-Scale CLIP-Seq and RNA-Seq Datasets.. Front Bioeng Biotechnol 2, 88 (2014).***\n",
    "\n",
    "***JH Li, S Liu, H Zhou, LH Qu, JH Yang. starBase v2.0: decoding miRNA-ceRNA, miRNA-ncRNA and protein-RNA interaction networks from large-scale CLIP-Seq data.. Nucleic Acids Res 42, D92-7 (2014).***\n",
    "\n",
    "- Comprehensive collection of lncRNA protein interaction from \n",
    "- Database dates from 2014 => contain datasets up to 2013 = not up to date.\n",
    "- No specific chromatin protein => have to be crossmatched with protein localization => Done from uniprot = few nuclei pure proteins\n",
    "\n",
    "**Cross-match Protein location Uniprot with data protein/lncRNA database from Starbase**\n",
    "\n",
    "Selection of a limited number of candidate CLIP-Seq datasets based on their localization and the experimental conditions:\n",
    "- Nuclear candidates: ALKBH5, ZC3H7B, FUS, TAF15, hnRNPC, DGCR8, FUS\n",
    "- Cytoplasmic candidates: AGO1, AGO3, AGO4, FXR1, LIN28B, PUM2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein</th>\n",
       "      <th>experiment</th>\n",
       "      <th>cell_line</th>\n",
       "      <th>replicate</th>\n",
       "      <th>treatment</th>\n",
       "      <th>localization</th>\n",
       "      <th>author</th>\n",
       "      <th>accession</th>\n",
       "      <th>filepath</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Dataset name</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>HPAR1</th>\n",
       "      <td>FXR1</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>1</td>\n",
       "      <td>wildtype FXR1, 4-thiouridine, RNase T1</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>ascano</td>\n",
       "      <td>GSE39682</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HPTA1</th>\n",
       "      <td>AGO1</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>1</td>\n",
       "      <td>4-thiouridine, RNase T1</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>hafner</td>\n",
       "      <td>GSE21918</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HPTA3</th>\n",
       "      <td>AGO3</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>1</td>\n",
       "      <td>4-thiouridine, RNase T1</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>hafner</td>\n",
       "      <td>GSE21918</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HPTA4</th>\n",
       "      <td>AGO4</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>1</td>\n",
       "      <td>4-thiouridine, RNase T1</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>hafner</td>\n",
       "      <td>GSE21918</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HPTPU</th>\n",
       "      <td>PUM2</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>1</td>\n",
       "      <td>4-thiouridine, RNase T1</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>hafner</td>\n",
       "      <td>GSE21918</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HPTLB</th>\n",
       "      <td>LIN28B</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>1</td>\n",
       "      <td>4-thiouridine, RNase T1</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>hafner</td>\n",
       "      <td>GSE44615</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HPRO6</th>\n",
       "      <td>AGO1</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>1</td>\n",
       "      <td>4-thiouridine, ML_MM_6</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>memczak</td>\n",
       "      <td>GSE43573</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HPRO7</th>\n",
       "      <td>AGO1</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>2</td>\n",
       "      <td>4-thiouridine, ML_MM_7</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>memczak</td>\n",
       "      <td>GSE43573</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HPRO8</th>\n",
       "      <td>AGO1</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>3</td>\n",
       "      <td>4-thiouridine, ML_MM_8</td>\n",
       "      <td>cytosol</td>\n",
       "      <td>memczak</td>\n",
       "      <td>GSE43573</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HPLAL</th>\n",
       "      <td>ALKBH5</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>1</td>\n",
       "      <td>4-thiouridine</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>baltz</td>\n",
       "      <td>GSE38201</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HPLZC</th>\n",
       "      <td>ZC3H7B</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>1</td>\n",
       "      <td>4-thiouridine</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>baltz</td>\n",
       "      <td>GSE38201</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HPHFW</th>\n",
       "      <td>FUS</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>1</td>\n",
       "      <td>4-thiouridine</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>hoell</td>\n",
       "      <td>SRA025082</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HPHTA</th>\n",
       "      <td>TAF15</td>\n",
       "      <td>PAR-CLIP</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>1</td>\n",
       "      <td>4-thiouridine</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>hoell</td>\n",
       "      <td>SRA025082</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HIUHN</th>\n",
       "      <td>hnRNPC</td>\n",
       "      <td>iCLIP</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>konig</td>\n",
       "      <td>SupplementaryData</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HHCD1</th>\n",
       "      <td>DGCR8</td>\n",
       "      <td>HITS-CLIP</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>1</td>\n",
       "      <td>D8 (endogenous DGCR8), replicate 1</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>macias</td>\n",
       "      <td>GSE39086</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HHCD2</th>\n",
       "      <td>DGCR8</td>\n",
       "      <td>HITS-CLIP</td>\n",
       "      <td>HEK293</td>\n",
       "      <td>2</td>\n",
       "      <td>D8 (endogenous DGCR8), replicate 2</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>macias</td>\n",
       "      <td>GSE39086</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HHMF1</th>\n",
       "      <td>FUS</td>\n",
       "      <td>HITS-CLIP</td>\n",
       "      <td>Brain</td>\n",
       "      <td>1</td>\n",
       "      <td>replicate1, RNase T1</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>nakaya</td>\n",
       "      <td>GSE43308</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HHMF2</th>\n",
       "      <td>FUS</td>\n",
       "      <td>HITS-CLIP</td>\n",
       "      <td>Brain</td>\n",
       "      <td>2</td>\n",
       "      <td>replicate2, RNase T1</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>nakaya</td>\n",
       "      <td>GSE43308</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HHMF3</th>\n",
       "      <td>FUS</td>\n",
       "      <td>HITS-CLIP</td>\n",
       "      <td>Brain</td>\n",
       "      <td>3</td>\n",
       "      <td>replicate3, RNase T1</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>nakaya</td>\n",
       "      <td>GSE43308</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HIUHC</th>\n",
       "      <td>hnRNPC</td>\n",
       "      <td>iCLIP</td>\n",
       "      <td>HeLa</td>\n",
       "      <td>1</td>\n",
       "      <td>untreated</td>\n",
       "      <td>nucleus</td>\n",
       "      <td>zarnack</td>\n",
       "      <td>E-MTAB-1371</td>\n",
       "      <td>/home/aleg/Data/Datasets/Epi2_pilot/Protein_in...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             protein experiment cell_line  replicate  \\\n",
       "Dataset name                                           \n",
       "HPAR1           FXR1   PAR-CLIP    HEK293          1   \n",
       "HPTA1           AGO1   PAR-CLIP    HEK293          1   \n",
       "HPTA3           AGO3   PAR-CLIP    HEK293          1   \n",
       "HPTA4           AGO4   PAR-CLIP    HEK293          1   \n",
       "HPTPU           PUM2   PAR-CLIP    HEK293          1   \n",
       "HPTLB         LIN28B   PAR-CLIP    HEK293          1   \n",
       "HPRO6           AGO1   PAR-CLIP    HEK293          1   \n",
       "HPRO7           AGO1   PAR-CLIP    HEK293          2   \n",
       "HPRO8           AGO1   PAR-CLIP    HEK293          3   \n",
       "HPLAL         ALKBH5   PAR-CLIP    HEK293          1   \n",
       "HPLZC         ZC3H7B   PAR-CLIP    HEK293          1   \n",
       "HPHFW            FUS   PAR-CLIP    HEK293          1   \n",
       "HPHTA          TAF15   PAR-CLIP    HEK293          1   \n",
       "HIUHN         hnRNPC      iCLIP      HeLa          1   \n",
       "HHCD1          DGCR8  HITS-CLIP    HEK293          1   \n",
       "HHCD2          DGCR8  HITS-CLIP    HEK293          2   \n",
       "HHMF1            FUS  HITS-CLIP     Brain          1   \n",
       "HHMF2            FUS  HITS-CLIP     Brain          2   \n",
       "HHMF3            FUS  HITS-CLIP     Brain          3   \n",
       "HIUHC         hnRNPC      iCLIP      HeLa          1   \n",
       "\n",
       "                                           treatment localization   author  \\\n",
       "Dataset name                                                                 \n",
       "HPAR1         wildtype FXR1, 4-thiouridine, RNase T1      cytosol   ascano   \n",
       "HPTA1                        4-thiouridine, RNase T1      cytosol   hafner   \n",
       "HPTA3                        4-thiouridine, RNase T1      cytosol   hafner   \n",
       "HPTA4                        4-thiouridine, RNase T1      cytosol   hafner   \n",
       "HPTPU                        4-thiouridine, RNase T1      cytosol   hafner   \n",
       "HPTLB                        4-thiouridine, RNase T1      cytosol   hafner   \n",
       "HPRO6                         4-thiouridine, ML_MM_6      cytosol  memczak   \n",
       "HPRO7                         4-thiouridine, ML_MM_7      cytosol  memczak   \n",
       "HPRO8                         4-thiouridine, ML_MM_8      cytosol  memczak   \n",
       "HPLAL                                  4-thiouridine      nucleus    baltz   \n",
       "HPLZC                                  4-thiouridine      nucleus    baltz   \n",
       "HPHFW                                  4-thiouridine      nucleus    hoell   \n",
       "HPHTA                                  4-thiouridine      nucleus    hoell   \n",
       "HIUHN                                              -      nucleus    konig   \n",
       "HHCD1             D8 (endogenous DGCR8), replicate 1      nucleus   macias   \n",
       "HHCD2             D8 (endogenous DGCR8), replicate 2      nucleus   macias   \n",
       "HHMF1                           replicate1, RNase T1      nucleus   nakaya   \n",
       "HHMF2                           replicate2, RNase T1      nucleus   nakaya   \n",
       "HHMF3                           replicate3, RNase T1      nucleus   nakaya   \n",
       "HIUHC                                      untreated      nucleus  zarnack   \n",
       "\n",
       "                      accession  \\\n",
       "Dataset name                      \n",
       "HPAR1                  GSE39682   \n",
       "HPTA1                  GSE21918   \n",
       "HPTA3                  GSE21918   \n",
       "HPTA4                  GSE21918   \n",
       "HPTPU                  GSE21918   \n",
       "HPTLB                  GSE44615   \n",
       "HPRO6                  GSE43573   \n",
       "HPRO7                  GSE43573   \n",
       "HPRO8                  GSE43573   \n",
       "HPLAL                  GSE38201   \n",
       "HPLZC                  GSE38201   \n",
       "HPHFW                 SRA025082   \n",
       "HPHTA                 SRA025082   \n",
       "HIUHN         SupplementaryData   \n",
       "HHCD1                  GSE39086   \n",
       "HHCD2                  GSE39086   \n",
       "HHMF1                  GSE43308   \n",
       "HHMF2                  GSE43308   \n",
       "HHMF3                  GSE43308   \n",
       "HIUHC               E-MTAB-1371   \n",
       "\n",
       "                                                       filepath  \n",
       "Dataset name                                                     \n",
       "HPAR1         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HPTA1         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HPTA3         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HPTA4         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HPTPU         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HPTLB         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HPRO6         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HPRO7         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HPRO8         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HPLAL         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HPLZC         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HPHFW         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HPHTA         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HIUHN         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HHCD1         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HHCD2         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HHMF1         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HHMF2         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HHMF3         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  \n",
       "HIUHC         /home/aleg/Data/Datasets/Epi2_pilot/Protein_in...  "
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = pd.read_csv(\"./Protein_interaction/Li_ClipSeq/Protocol_info/Selected_datasets.csv\", sep=\"\\t\", index_col=0)\n",
    "df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Convert coordinates from hg19 to hg39 with Crossmap and remap on last gencode human genes...\n",
    "\n",
    "#### Coordinate conversion using CrossMap and a hg19 tp hg38 chain file in BASH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1\t758393\t758417\tHHCD1_1_DGCR8_D8.1_3\t9\t+\n",
      "chr1\t762871\t762896\tHHCD1_2392_DGCR8_D8.1_5\t10\t-\n",
      "chr1\t876695\t876718\tHHCD1_2_DGCR8_D8.1_4\t4\t+\n",
      "\n",
      "chr1\t823013\t823037\tHHCD1_1_DGCR8_D8.1_3\t9\t+\n",
      "chr1\t827491\t827516\tHHCD1_2392_DGCR8_D8.1_5\t10\t-\n",
      "chr1\t941315\t941338\tHHCD1_2_DGCR8_D8.1_4\t4\t+\n",
      "\n",
      "chr2\t242747383\t242747404\tHHCD1_30364_DGCR8_D8.1_1\t14\t-\tFail\n",
      "\n",
      "Only 2 lines in the file\n",
      "\n"
     ]
    }
   ],
   "source": [
    "def liftover (input_file, output_file, unmap_file, liftover_chainfile):\n",
    "    # Conversion of coordinates with Crossmap/liftover\n",
    "    \n",
    "    with open (output_file, \"w\") as output, open(unmap_file, \"w\") as unmap:\n",
    "        # Build the command line\n",
    "        cmd = \"CrossMap.py bed {} {}\".format(liftover_chainfile, input_file)\n",
    "        # Write Liftovers coordinates\n",
    "        for line in pycl.bash(cmd, ret_stdout=True).split(\"\\n\"):\n",
    "            ls = line.split(\"\\t\")\n",
    "            try:\n",
    "                output.write(\"{}\\t{}\\t{}\\t{}\\t{}\\t{}\\n\".format(ls[7], ls[8], ls[9], ls[10], ls[11], ls[12]))\n",
    "            except IndexError as E:\n",
    "                unmap.write(line+\"\\n\")\n",
    "\n",
    "# Test with one file\n",
    "pycl.mkdir(\"./Protein_interaction/Li_ClipSeq/test\")\n",
    "liftover_chainfile = \"../../LiftOver_chain_files/hg19ToHg38.over.chain.gz\"\n",
    "input_file = \"./Protein_interaction/Li_ClipSeq/Human/HHCD1.bed\"\n",
    "output_file = \"./Protein_interaction/Li_ClipSeq/test/HHCD1_converted.bed\"\n",
    "unmap_file = \"./Protein_interaction/Li_ClipSeq/test/HHCD1_unmap.bed\"\n",
    "liftover(input_file, output_file, unmap_file, liftover_chainfile)\n",
    "pycl.head(input_file, 3)\n",
    "pycl.head(output_file, 3)\n",
    "pycl.head(unmap_file, 3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "#### Remap on the human transcriptome with pyBedTools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\tchr1\t823013\t823037\tHHCD1_1_DGCR8_D8.1_3\t9\t+\n",
      "1\tchr1\t827491\t827516\tHHCD1_2392_DGCR8_D8.1_5\t10\t-\n",
      "2\tchr1\t941315\t941338\tHHCD1_2_DGCR8_D8.1_4\t4\t+\n",
      "\n",
      "58711\tchrX\t155605820\t155605843\tHHCD1_57427_DGCR8_D8.1_3\t12\t+\n",
      "58712\tchrX\t155612161\t155612184\tHHCD1_58704_DGCR8_D8.1_3\t2\t-\n",
      "58713\tchrX\t155615409\t155615433\tHHCD1_57428_DGCR8_D8.1_3\t4\t+\n",
      "\n",
      "0\tchrom\tstart\tend\tgene_id|gene_name|gene_type|len\tscore\tstrand\n",
      "1\tchr1\t827491\t827516\tENSG00000225880.5|LINC00115|lincRNA|25\t10\t-\n",
      "2\tchr1\t827491\t827516\tENSG00000230021.8|RP5-857K21.4|lincRNA|25\t10\t-\n",
      "\n",
      "19823\tchrX\t155227749\t155227778\tENSG00000155959.10|VBP1|protein_coding|29\t71\t+\n",
      "19824\tchrX\t155605820\t155605843\tENSG00000224533.4|TMLHE-AS1|antisense|23\t12\t+\n",
      "19825\tchrX\t155612161\t155612184\tENSG00000185973.10|TMLHE|protein_coding|23\t2\t-\n",
      "\n"
     ]
    }
   ],
   "source": [
    "import pybedtools\n",
    "\n",
    "def intersect_bed (annotation_file, peak_file, output_file):\n",
    "    \n",
    "    # Parse file with pybedtools and intersect the coordinate ranges\n",
    "    peak = pybedtools.BedTool(peak_file)\n",
    "    annotation = pybedtools.BedTool(annotation_file)\n",
    "    intersection = peak.intersect(annotation, wo=True, s=True)\n",
    "    \n",
    "    # Reformat the file generated by pybedtools with the merged intervals to a simple Bed format\n",
    "    init_template=[\"{chrom}\",\"\\t\",\"{start}\",\"\\t\",\"{end}\",\"\\t\",\"{info}\",\"\\t\",\"{score}\",\"\\t\",\"{strand}\",\"\\t\",\"{chrom2}\",\"\\t\",\"{source}\",\"\\t\",\"{feature}\",\"\\t\",\"{start2}\",\n",
    "                   \"\\t\",\"{end2}\",\"\\t\",\"{score2}\",\"\\t\",\"{strand2}\",\"\\t\",\"{frame}\",\"\\tID=\",\"{ID}\",\";gene_id=\",\"{gene_id}\",\";gene_type=\",\"{gene_type}\",\n",
    "                   \";gene_status=\",\"{gene_status}\",\";gene_name=\",\"{gene_name}\",\";level=\",\"{level}\",\";havana_gene=\",\"{havana_gene}\",\"\\t\",\"{len}\"]\n",
    "    \n",
    "    final_template=[\"{chrom}\",\"\\t\",\"{start}\",\"\\t\",\"{end}\",\"\\t\",\"{gene_id}\",\"|\",\"{gene_name}\",\"|\",\"{gene_type}\",\"|\",\"{len}\",\"\\t\",\"{score}\",\"\\t\",\"{strand}\"]\n",
    "\n",
    "    pycl.reformat_table(\n",
    "        input_file=intersection.fn, output_file=output_file, init_template=init_template, final_template=final_template,\n",
    "        replace_internal_space='_', replace_null_val=\"-\", keep_original_header=False, predicate=lambda v: v[\"feature\"] == \"gene\", header_from_final_template=True)\n",
    "\n",
    "# Test with one file\n",
    "annotation_file = \"../../Reference_Annotation/gencode_v24.gff3\"\n",
    "peak_file = \"./Protein_interaction/Li_ClipSeq/test/HHCD1_converted.bed\"\n",
    "output_file = \"./Protein_interaction/Li_ClipSeq/test/HHCD1_annotated.bed\"\n",
    "intersect_bed (annotation_file, peak_file, output_file)\n",
    "pycl.linerange(peak_file)\n",
    "pycl.linerange(output_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### List the uniq gene ids for each samples\n",
    "\n",
    "for experiment with several samples, the function only retain those in **common in ALL the replicates**"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\tgene_id\tgene_name\tgene_type\tall_count\trep_count\n",
      "1\tENSG00000141101.12\tNOB1\tprotein_coding\t7\t1\n",
      "2\tENSG00000170677.5\tSOCS6\tprotein_coding\t1\t1\n",
      "\n",
      "8141\tENSG00000135829.16\tDHX9\tprotein_coding\t8\t1\n",
      "8142\tENSG00000169223.14\tLMAN2\tprotein_coding\t1\t1\n",
      "8143\tENSG00000149925.16\tALDOA\tprotein_coding\t11\t1\n",
      "\n",
      "0\tgene_id\tgene_name\tgene_type\tall_count\trep_count\n",
      "1\tENSG00000170677.5\tSOCS6\tprotein_coding\t2\t2\n",
      "2\tENSG00000034053.14\tAPBA2\tprotein_coding\t5\t2\n",
      "\n",
      "3250\tENSG00000258472.8\tRP11-192H23.4\tprotein_coding\t18\t2\n",
      "3251\tENSG00000105662.15\tCRTC1\tprotein_coding\t2\t2\n",
      "3252\tENSG00000169223.14\tLMAN2\tprotein_coding\t3\t2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "def unique_genes_list (input_file_list, output_file):\n",
    "\n",
    "    gene_dict = {}\n",
    "\n",
    "    # iterate on replicates of the same experiment\n",
    "    for input_file in input_file_list:\n",
    "        df = pd.read_csv(input_file, sep=\"[\\t|]\", engine='python')\n",
    "\n",
    "        # iterate on lines of a replicate bed file\n",
    "        for gene_id, gene_df in df.groupby(\"gene_id\"):\n",
    "            if gene_id not in gene_dict:\n",
    "                gene_dict[gene_id] = {\n",
    "                    \"gene_name\":gene_df[\"gene_name\"].unique()[0],\n",
    "                    \"gene_type\":gene_df[\"gene_type\"].unique()[0],\n",
    "                    \"all_count\":0, \"rep_count\":0}\n",
    "            gene_dict[gene_id][\"all_count\"]+=len(gene_df)\n",
    "            gene_dict[gene_id][\"rep_count\"]+=1\n",
    "\n",
    "    # Write the valid gene entries in a file \n",
    "    with open (output_file, \"w\") as output:\n",
    "        output.write(\"gene_id\\tgene_name\\tgene_type\\tall_count\\trep_count\\n\")\n",
    "        for gene_id, val in gene_dict.items():\n",
    "            if val[\"rep_count\"] == len(input_file_list):\n",
    "                output.write(\"{}\\t{}\\t{}\\t{}\\t{}\\n\".format(gene_id, val[\"gene_name\"], val[\"gene_type\"], val[\"all_count\"], val[\"rep_count\"]))\n",
    "            \n",
    "# Test with one file\n",
    "input_file = [\"./Protein_interaction/Li_ClipSeq/test/HCTA1_annotated.bed\"]\n",
    "output_file = \"./Protein_interaction/Li_ClipSeq/test/HCTA1_uniq_gene.tsv\"\n",
    "unique_genes_list(input_file, output_file)\n",
    "pycl.linerange(output_file)\n",
    "\n",
    "# Test with a list of files\n",
    "input_file = [ \"./Protein_interaction/Li_ClipSeq/test/HCTA1_annotated.bed\", \"./Protein_interaction/Li_ClipSeq/test/HHCD1_annotated.bed\"]\n",
    "output_file = \"./Protein_interaction/Li_ClipSeq/test/HHMF1-3_uniq_gene.tsv\"\n",
    "unique_genes_list(input_file, output_file)\n",
    "pycl.linerange(output_file)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Merge the replicates of the same experiment "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Iterate on all the files listed in the sample list and apply hg38 coordinate conversion, gencodev24 annotation and listing of unique genes  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis experiment  AGO1_PAR-CLIP_HEK293_cytosol_hafner\n",
      "\tAnalysis dataset HPTA1\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tList uniq genes\n",
      "Analysis experiment  AGO1_PAR-CLIP_HEK293_cytosol_memczak\n",
      "\tAnalysis dataset HPRO6\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tAnalysis dataset HPRO7\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tAnalysis dataset HPRO8\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tList uniq genes\n",
      "Analysis experiment  AGO3_PAR-CLIP_HEK293_cytosol_hafner\n",
      "\tAnalysis dataset HPTA3\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tList uniq genes\n",
      "Analysis experiment  AGO4_PAR-CLIP_HEK293_cytosol_hafner\n",
      "\tAnalysis dataset HPTA4\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tList uniq genes\n",
      "Analysis experiment  ALKBH5_PAR-CLIP_HEK293_nucleus_baltz\n",
      "\tAnalysis dataset HPLAL\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tList uniq genes\n",
      "Analysis experiment  DGCR8_HITS-CLIP_HEK293_nucleus_macias\n",
      "\tAnalysis dataset HHCD1\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tAnalysis dataset HHCD2\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tList uniq genes\n",
      "Analysis experiment  FUS_HITS-CLIP_Brain_nucleus_nakaya\n",
      "\tAnalysis dataset HHMF1\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tAnalysis dataset HHMF2\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tAnalysis dataset HHMF3\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tList uniq genes\n",
      "Analysis experiment  FUS_PAR-CLIP_HEK293_nucleus_hoell\n",
      "\tAnalysis dataset HPHFW\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tList uniq genes\n",
      "Analysis experiment  FXR1_PAR-CLIP_HEK293_cytosol_ascano\n",
      "\tAnalysis dataset HPAR1\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tList uniq genes\n",
      "Analysis experiment  LIN28B_PAR-CLIP_HEK293_cytosol_hafner\n",
      "\tAnalysis dataset HPTLB\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tList uniq genes\n",
      "Analysis experiment  PUM2_PAR-CLIP_HEK293_cytosol_hafner\n",
      "\tAnalysis dataset HPTPU\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tList uniq genes\n",
      "Analysis experiment  TAF15_PAR-CLIP_HEK293_nucleus_hoell\n",
      "\tAnalysis dataset HPHTA\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tList uniq genes\n",
      "Analysis experiment  ZC3H7B_PAR-CLIP_HEK293_nucleus_baltz\n",
      "\tAnalysis dataset HPLZC\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tList uniq genes\n",
      "Analysis experiment  hnRNPC_iCLIP_HeLa_nucleus_konig\n",
      "\tAnalysis dataset HIUHN\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tList uniq genes\n",
      "Analysis experiment  hnRNPC_iCLIP_HeLa_nucleus_zarnack\n",
      "\tAnalysis dataset HIUHC\n",
      "\t\tConverting coordinates with liftover\n",
      "\t\tIntersect with annotation file\n",
      "\tList uniq genes\n"
     ]
    }
   ],
   "source": [
    "liftover_chainfile = \"../../LiftOver_chain_files/hg19ToHg38.over.chain.gz\"\n",
    "annotation_file = \"../../Reference_Annotation/gencode_v24.gff3\"\n",
    "intermediate_folder = \"./Protein_interaction/Li_ClipSeq/selected_datasets/\"\n",
    "final_folder = \"./Protein_interaction/Li_ClipSeq/final_datasets/\"\n",
    "pycl.mkdir(intermediate_folder)\n",
    "pycl.mkdir(final_folder)\n",
    "\n",
    "# df containing the sample list details\n",
    "df = pd.read_csv(\"./Protein_interaction/Li_ClipSeq/Protocol_info/Selected_datasets.csv\", sep=\"\\t\", index_col=0)\n",
    "\n",
    "# dict to count the file of the different dataset and experiments\n",
    "dataset_dict = {}\n",
    "exp_dict = {}\n",
    "\n",
    "# Iterate by all sample of the same esperiment (include replicates if there are some replicates)\n",
    "for (prot,exp, cl, loc, aut), exp_df in df.groupby([\"protein\",\"experiment\", \"cell_line\", \"localization\", \"author\"]):\n",
    "    exp_id = \"{}_{}_{}_{}_{}\".format(prot,exp, cl, loc, aut)\n",
    "    print(\"Analysis experiment \", exp_id)\n",
    "    intersect_file_list=[]\n",
    "    \n",
    "    # Iterate on the replicates of the same experiment\n",
    "    for dataset_id, line in exp_df.iterrows():\n",
    "\n",
    "        print(\"\\tAnalysis dataset {}\".format(dataset_id))\n",
    "        input_file = line[\"filepath\"]\n",
    "        \n",
    "        print (\"\\t\\tConverting coordinates with liftover\")\n",
    "        converted_file = \"{}{}_coord_hg38.bed\".format(intermediate_folder, dataset_id) \n",
    "        unmap_file = \"{}{}_unmap.bed\".format(intermediate_folder, dataset_id) \n",
    "        liftover(input_file, converted_file, unmap_file, liftover_chainfile)\n",
    "\n",
    "        print (\"\\t\\tIntersect with annotation file\")\n",
    "        intersect_file = \"{}{}_coord_hg38_annot_gencode24.bed\".format(intermediate_folder, dataset_id)\n",
    "        intersect_file_list.append (intersect_file)\n",
    "        intersect_bed (annotation_file, converted_file, intersect_file)\n",
    "        \n",
    "        # Fill the counter\n",
    "        dataset_dict[dataset_id] = {\n",
    "            \"Init_sites\":pycl.simplecount(input_file),\n",
    "            \"Converted_coord_sites\": pycl.simplecount(converted_file),\n",
    "            \"Unmap_sites\": pycl.simplecount(unmap_file),\n",
    "            \"Annotated_sites\": pycl.simplecount(intersect_file)}\n",
    "    \n",
    "    # list uniq genes for each experiment. Replicate experiment and inner-joined before    \n",
    "    print (\"\\tList uniq genes\")\n",
    "    unique_gene_file = \"{}{}_uniq_genes.tsv\".format(final_folder, exp_id) \n",
    "    unique_genes_list (intersect_file_list, unique_gene_file)\n",
    "    exp_dict[exp_id] = {\n",
    "        \"Uniq_genes\": pycl.simplecount(unique_gene_file),\n",
    "        \"Replicate_Number\": len(intersect_file_list)}\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table width=100%><tr><td> </td><td><b>Annotated_sites</b></td><td><b>Converted_coord_sites</b></td><td><b>Init_sites</b></td><td><b>Unmap_sites</b></td></tr><tr><td><b>HHCD1</b></td><td>19826</td><td>58714</td><td>58704</td><td>2</td></tr><tr><td><b>HHCD2</b></td><td>76848</td><td>124378</td><td>124354</td><td>5</td></tr><tr><td><b>HHMF1</b></td><td>191972</td><td>325659</td><td>326115</td><td>546</td></tr><tr><td><b>HHMF2</b></td><td>258589</td><td>414603</td><td>415091</td><td>624</td></tr><tr><td><b>HHMF3</b></td><td>269695</td><td>471039</td><td>471590</td><td>681</td></tr><tr><td><b>HIUHC</b></td><td>429723</td><td>438395</td><td>438360</td><td>30</td></tr><tr><td><b>HIUHN</b></td><td>23473</td><td>24446</td><td>24448</td><td>5</td></tr><tr><td><b>HPAR1</b></td><td>8405</td><td>9017</td><td>9017</td><td>2</td></tr><tr><td><b>HPHFW</b></td><td>37719</td><td>39988</td><td>39983</td><td>1</td></tr><tr><td><b>HPHTA</b></td><td>8582</td><td>8678</td><td>8677</td><td>1</td></tr><tr><td><b>HPLAL</b></td><td>1632</td><td>1597</td><td>1596</td><td>1</td></tr><tr><td><b>HPLZC</b></td><td>33465</td><td>32805</td><td>32798</td><td>1</td></tr><tr><td><b>HPRO6</b></td><td>78105</td><td>82219</td><td>82196</td><td>3</td></tr><tr><td><b>HPRO7</b></td><td>2367</td><td>3454</td><td>3453</td><td>1</td></tr><tr><td><b>HPRO8</b></td><td>14050</td><td>15890</td><td>15887</td><td>2</td></tr><tr><td><b>HPTA1</b></td><td>11428</td><td>12165</td><td>12163</td><td>1</td></tr><tr><td><b>HPTA3</b></td><td>18576</td><td>20006</td><td>20001</td><td>1</td></tr><tr><td><b>HPTA4</b></td><td>2895</td><td>4398</td><td>4397</td><td>1</td></tr><tr><td><b>HPTLB</b></td><td>28249</td><td>31802</td><td>31796</td><td>2</td></tr><tr><td><b>HPTPU</b></td><td>7746</td><td>7524</td><td>7523</td><td>1</td></tr></table>"
      ],
      "text/plain": [
       "dict_to_html([('HHCD1',\n",
       "               {'Annotated_sites': 19826,\n",
       "                'Converted_coord_sites': 58714,\n",
       "                'Init_sites': 58704,\n",
       "                'Unmap_sites': 2}),\n",
       "              ('HHCD2',\n",
       "               {'Annotated_sites': 76848,\n",
       "                'Converted_coord_sites': 124378,\n",
       "                'Init_sites': 124354,\n",
       "                'Unmap_sites': 5}),\n",
       "              ('HHMF1',\n",
       "               {'Annotated_sites': 191972,\n",
       "                'Converted_coord_sites': 325659,\n",
       "                'Init_sites': 326115,\n",
       "                'Unmap_sites': 546}),\n",
       "              ('HHMF2',\n",
       "               {'Annotated_sites': 258589,\n",
       "                'Converted_coord_sites': 414603,\n",
       "                'Init_sites': 415091,\n",
       "                'Unmap_sites': 624}),\n",
       "              ('HHMF3',\n",
       "               {'Annotated_sites': 269695,\n",
       "                'Converted_coord_sites': 471039,\n",
       "                'Init_sites': 471590,\n",
       "                'Unmap_sites': 681}),\n",
       "              ('HIUHC',\n",
       "               {'Annotated_sites': 429723,\n",
       "                'Converted_coord_sites': 438395,\n",
       "                'Init_sites': 438360,\n",
       "                'Unmap_sites': 30}),\n",
       "              ('HIUHN',\n",
       "               {'Annotated_sites': 23473,\n",
       "                'Converted_coord_sites': 24446,\n",
       "                'Init_sites': 24448,\n",
       "                'Unmap_sites': 5}),\n",
       "              ('HPAR1',\n",
       "               {'Annotated_sites': 8405,\n",
       "                'Converted_coord_sites': 9017,\n",
       "                'Init_sites': 9017,\n",
       "                'Unmap_sites': 2}),\n",
       "              ('HPHFW',\n",
       "               {'Annotated_sites': 37719,\n",
       "                'Converted_coord_sites': 39988,\n",
       "                'Init_sites': 39983,\n",
       "                'Unmap_sites': 1}),\n",
       "              ('HPHTA',\n",
       "               {'Annotated_sites': 8582,\n",
       "                'Converted_coord_sites': 8678,\n",
       "                'Init_sites': 8677,\n",
       "                'Unmap_sites': 1}),\n",
       "              ('HPLAL',\n",
       "               {'Annotated_sites': 1632,\n",
       "                'Converted_coord_sites': 1597,\n",
       "                'Init_sites': 1596,\n",
       "                'Unmap_sites': 1}),\n",
       "              ('HPLZC',\n",
       "               {'Annotated_sites': 33465,\n",
       "                'Converted_coord_sites': 32805,\n",
       "                'Init_sites': 32798,\n",
       "                'Unmap_sites': 1}),\n",
       "              ('HPRO6',\n",
       "               {'Annotated_sites': 78105,\n",
       "                'Converted_coord_sites': 82219,\n",
       "                'Init_sites': 82196,\n",
       "                'Unmap_sites': 3}),\n",
       "              ('HPRO7',\n",
       "               {'Annotated_sites': 2367,\n",
       "                'Converted_coord_sites': 3454,\n",
       "                'Init_sites': 3453,\n",
       "                'Unmap_sites': 1}),\n",
       "              ('HPRO8',\n",
       "               {'Annotated_sites': 14050,\n",
       "                'Converted_coord_sites': 15890,\n",
       "                'Init_sites': 15887,\n",
       "                'Unmap_sites': 2}),\n",
       "              ('HPTA1',\n",
       "               {'Annotated_sites': 11428,\n",
       "                'Converted_coord_sites': 12165,\n",
       "                'Init_sites': 12163,\n",
       "                'Unmap_sites': 1}),\n",
       "              ('HPTA3',\n",
       "               {'Annotated_sites': 18576,\n",
       "                'Converted_coord_sites': 20006,\n",
       "                'Init_sites': 20001,\n",
       "                'Unmap_sites': 1}),\n",
       "              ('HPTA4',\n",
       "               {'Annotated_sites': 2895,\n",
       "                'Converted_coord_sites': 4398,\n",
       "                'Init_sites': 4397,\n",
       "                'Unmap_sites': 1}),\n",
       "              ('HPTLB',\n",
       "               {'Annotated_sites': 28249,\n",
       "                'Converted_coord_sites': 31802,\n",
       "                'Init_sites': 31796,\n",
       "                'Unmap_sites': 2}),\n",
       "              ('HPTPU',\n",
       "               {'Annotated_sites': 7746,\n",
       "                'Converted_coord_sites': 7524,\n",
       "                'Init_sites': 7523,\n",
       "                'Unmap_sites': 1})])"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pycl.dict_to_html(dataset_dict)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table width=100%><tr><td> </td><td><b>Replicate_Number</b></td><td><b>Uniq_genes</b></td></tr><tr><td><b>AGO1_PAR-CLIP_HEK293_cytosol_hafner</b></td><td>1</td><td>5962</td></tr><tr><td><b>AGO1_PAR-CLIP_HEK293_cytosol_memczak</b></td><td>3</td><td>1603</td></tr><tr><td><b>AGO3_PAR-CLIP_HEK293_cytosol_hafner</b></td><td>1</td><td>8181</td></tr><tr><td><b>AGO4_PAR-CLIP_HEK293_cytosol_hafner</b></td><td>1</td><td>2337</td></tr><tr><td><b>ALKBH5_PAR-CLIP_HEK293_nucleus_baltz</b></td><td>1</td><td>1184</td></tr><tr><td><b>DGCR8_HITS-CLIP_HEK293_nucleus_macias</b></td><td>2</td><td>8073</td></tr><tr><td><b>FUS_HITS-CLIP_Brain_nucleus_nakaya</b></td><td>3</td><td>18024</td></tr><tr><td><b>FUS_PAR-CLIP_HEK293_nucleus_hoell</b></td><td>1</td><td>8310</td></tr><tr><td><b>FXR1_PAR-CLIP_HEK293_cytosol_ascano</b></td><td>1</td><td>2824</td></tr><tr><td><b>LIN28B_PAR-CLIP_HEK293_cytosol_hafner</b></td><td>1</td><td>10080</td></tr><tr><td><b>PUM2_PAR-CLIP_HEK293_cytosol_hafner</b></td><td>1</td><td>4457</td></tr><tr><td><b>TAF15_PAR-CLIP_HEK293_nucleus_hoell</b></td><td>1</td><td>3662</td></tr><tr><td><b>ZC3H7B_PAR-CLIP_HEK293_nucleus_baltz</b></td><td>1</td><td>7329</td></tr><tr><td><b>hnRNPC_iCLIP_HeLa_nucleus_konig</b></td><td>1</td><td>5869</td></tr><tr><td><b>hnRNPC_iCLIP_HeLa_nucleus_zarnack</b></td><td>1</td><td>13029</td></tr></table>"
      ],
      "text/plain": [
       "dict_to_html([('AGO1_PAR-CLIP_HEK293_cytosol_hafner',\n",
       "               {'Replicate_Number': 1, 'Uniq_genes': 5962}),\n",
       "              ('AGO1_PAR-CLIP_HEK293_cytosol_memczak',\n",
       "               {'Replicate_Number': 3, 'Uniq_genes': 1603}),\n",
       "              ('AGO3_PAR-CLIP_HEK293_cytosol_hafner',\n",
       "               {'Replicate_Number': 1, 'Uniq_genes': 8181}),\n",
       "              ('AGO4_PAR-CLIP_HEK293_cytosol_hafner',\n",
       "               {'Replicate_Number': 1, 'Uniq_genes': 2337}),\n",
       "              ('ALKBH5_PAR-CLIP_HEK293_nucleus_baltz',\n",
       "               {'Replicate_Number': 1, 'Uniq_genes': 1184}),\n",
       "              ('DGCR8_HITS-CLIP_HEK293_nucleus_macias',\n",
       "               {'Replicate_Number': 2, 'Uniq_genes': 8073}),\n",
       "              ('FUS_HITS-CLIP_Brain_nucleus_nakaya',\n",
       "               {'Replicate_Number': 3, 'Uniq_genes': 18024}),\n",
       "              ('FUS_PAR-CLIP_HEK293_nucleus_hoell',\n",
       "               {'Replicate_Number': 1, 'Uniq_genes': 8310}),\n",
       "              ('FXR1_PAR-CLIP_HEK293_cytosol_ascano',\n",
       "               {'Replicate_Number': 1, 'Uniq_genes': 2824}),\n",
       "              ('LIN28B_PAR-CLIP_HEK293_cytosol_hafner',\n",
       "               {'Replicate_Number': 1, 'Uniq_genes': 10080}),\n",
       "              ('PUM2_PAR-CLIP_HEK293_cytosol_hafner',\n",
       "               {'Replicate_Number': 1, 'Uniq_genes': 4457}),\n",
       "              ('TAF15_PAR-CLIP_HEK293_nucleus_hoell',\n",
       "               {'Replicate_Number': 1, 'Uniq_genes': 3662}),\n",
       "              ('ZC3H7B_PAR-CLIP_HEK293_nucleus_baltz',\n",
       "               {'Replicate_Number': 1, 'Uniq_genes': 7329}),\n",
       "              ('hnRNPC_iCLIP_HeLa_nucleus_konig',\n",
       "               {'Replicate_Number': 1, 'Uniq_genes': 5869}),\n",
       "              ('hnRNPC_iCLIP_HeLa_nucleus_zarnack',\n",
       "               {'Replicate_Number': 1, 'Uniq_genes': 13029})])"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pycl.dict_to_html(exp_dict)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "### Further dataset analysis\n",
    "\n",
    "**It might also be interesting to have a set of core generic RNA associated with nuclear or cytoplasmic candidates. To do so, I need to intersect the data irrespective of their cell line**\n",
    "**I will create several datasets**\n",
    "\n",
    "* RNA found only in at least one nuclear proteins but no cytoplasmic proteins\n",
    "* RNA found only in at least one cytoplasmic proteins but no nuclear proteins\n",
    "* RNA found in all nuclear proteins but no cytoplasmic protein\n",
    "* RNA found in all cytoplasmic proteins but no nuclear protein"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\tgene_id\tgene_name\tgene_type\tall_count\trep_count\n",
      "1\tENSG00000137807.13\tKIF23\tprotein_coding\t2\t1\n",
      "2\tENSG00000272305.5\tRP11-894J14.5\tprotein_coding\t1\t1\n",
      "\n",
      "5959\tENSG00000247315.3\tZCCHC3\tprotein_coding\t2\t1\n",
      "5960\tENSG00000105401.6\tCDC37\tprotein_coding\t1\t1\n",
      "5961\tENSG00000109586.11\tGALNT7\tprotein_coding\t2\t1\n",
      "\n"
     ]
    }
   ],
   "source": [
    "pycl.linerange(\"./Protein_interaction/Li_ClipSeq/final_datasets/AGO1_PAR-CLIP_HEK293_cytosol_hafner_uniq_genes.tsv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Analysis dataset  TAF15_PAR-CLIP_HEK293_nucleus_hoell_uniq_genes\n",
      "Analysis dataset  AGO4_PAR-CLIP_HEK293_cytosol_hafner_uniq_genes\n",
      "Analysis dataset  FUS_PAR-CLIP_HEK293_nucleus_hoell_uniq_genes\n",
      "Analysis dataset  ALKBH5_PAR-CLIP_HEK293_nucleus_baltz_uniq_genes\n",
      "Analysis dataset  hnRNPC_iCLIP_HeLa_nucleus_zarnack_uniq_genes\n",
      "Analysis dataset  AGO1_PAR-CLIP_HEK293_cytosol_memczak_uniq_genes\n",
      "Analysis dataset  AGO3_PAR-CLIP_HEK293_cytosol_hafner_uniq_genes\n",
      "Analysis dataset  LIN28B_PAR-CLIP_HEK293_cytosol_hafner_uniq_genes\n",
      "Analysis dataset  ZC3H7B_PAR-CLIP_HEK293_nucleus_baltz_uniq_genes\n",
      "Analysis dataset  hnRNPC_iCLIP_HeLa_nucleus_konig_uniq_genes\n",
      "Analysis dataset  AGO1_PAR-CLIP_HEK293_cytosol_hafner_uniq_genes\n",
      "Analysis dataset  FUS_HITS-CLIP_Brain_nucleus_nakaya_uniq_genes\n",
      "Analysis dataset  PUM2_PAR-CLIP_HEK293_cytosol_hafner_uniq_genes\n",
      "Analysis dataset  DGCR8_HITS-CLIP_HEK293_nucleus_macias_uniq_genes\n",
      "Analysis dataset  FXR1_PAR-CLIP_HEK293_cytosol_ascano_uniq_genes\n",
      "0\tgene_id\tgene_name\tgene_type\tcytosol_protein\tnuclear_protein\tnuclearness\n",
      "1\tENSG00000247877.6\tCTD-2001C12.1\tlincRNA\t0\t1\t1\n",
      "2\tENSG00000267053.6\tCTD-3162L10.1\tlincRNA\t0\t1\t1\n",
      "\n",
      "24195\tENSG00000066027.11\tPPP2R5A\tprotein_coding\t3\t3\t0\n",
      "24196\tENSG00000249115.8\tHAUS5\tprotein_coding\t1\t3\t2\n",
      "24197\tENSG00000155846.16\tPPARGC1B\tprotein_coding\t7\t5\t-2\n",
      "\n"
     ]
    }
   ],
   "source": [
    "input_folder = \"./Protein_interaction/Li_ClipSeq/final_datasets/\"\n",
    "output_file = \"./Protein_interaction/Li_ClipSeq/final_datasets/ALL_dataset_uniq_genes.tsv\"\n",
    "remove(output_file)\n",
    "# dict to count the file of the different categorie\n",
    "gene_loc_dict = {}\n",
    "\n",
    "for input_file in glob(input_folder+\"*tsv\"):\n",
    "    sample_name = pycl.file_basename(input_file)\n",
    "    print(\"Analysis dataset \", sample_name)\n",
    "    \n",
    "    localization = sample_name.split(\"_\")[3]\n",
    "    unique_gene_df = pd.read_csv(input_file, sep=\"\\t\", index_col=0)\n",
    "    \n",
    "    for gene_id, gene_line in unique_gene_df.iterrows():\n",
    "        if gene_id not in gene_loc_dict:\n",
    "            gene_loc_dict[gene_id] = {\"gene_name\": gene_line[\"gene_name\"] ,\"gene_type\":gene_line[\"gene_type\"] ,\"cytosol\":0, \"nucleus\":0}\n",
    "        gene_loc_dict[gene_id][localization] += 1\n",
    "\n",
    "# Write results in a file\n",
    "with open (output_file, \"w\") as output:\n",
    "    output.write(\"gene_id\\tgene_name\\tgene_type\\tcytosol_protein\\tnuclear_protein\\tnuclearness\\n\")\n",
    "    for gene_id, val in gene_loc_dict.items():\n",
    "        output.write(\"{}\\t{}\\t{}\\t{}\\t{}\\t{}\\n\".format(gene_id,val[\"gene_name\"],val[\"gene_type\"],val[\"cytosol\"],val[\"nucleus\"],val[\"nucleus\"]-val[\"cytosol\"]))\n",
    "\n",
    "pycl.linerange(output_file)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAABrgAAAJSCAYAAACYxsVJAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJzs3X+s73dd2PHnpRfLKLeAegekbrgsy3sx+2fGSVJg5Ydg\n3CJuAaKhYUj9g5DFNCwugWQoIxshbpplmmVJsdIwXFBTnJ0ZaZsxBGWriYniH3s7MbhJkV4Wtt4q\nF3t77/64p0u9Fu5pueX01ft4JDf9nvf38z33/Tm5r7+efX/OsfPnzwcAAAAAAABTPO2oNwAAAAAA\nAACPhcAFAAAAAADAKAIXAAAAAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCjHD3PRWuvG\n6h9XD1Y/Wn2q+kAXAtnnqjfuvR88uO7m6qHqlr33rWut49X7qxdWZ6s3770/c5nvAwAAAAAAgCvE\nsfPnz3/VC9Za31h9svqb1Ynq3dXTq/+49759rfXPq//ZheD1m9V3dCFk/Ub10uo11d/ae//wWutV\n1Q/tvX/gCbofAAAAAAAAnuIOc4Lru6q79t5/Uv1J9Za11u9Xbzl4/47qR6rfre7Zez9Qtdb6RPWS\n6pXVbQfX3l3devm2DwAAAAAAwJXmML+D61ura9Za/2Gt9bG11iuqZ+69Hzx4/77qBdXzqlOP+Nyp\ni9f33uercwePLQQAAAAAAIDH7DCh6Vj1jdXf70Ls+ujB2iPf/0qfezSHiWoAAAAAAADwqA4TuD5f\n/fre+1z1+2ut09WDa62r995frq6rPlvd24UTWw+7rgu/u+ve6vnVpx4+ubX3PvvV/sLz58+fP3bs\nK/UxAAAAAAAAnsIuGYkOE7jurH52rfXjXTjJ9azqI9Xrqg9Wrz34+p7qfWuta6tz1fXVzdWzq9dX\nd1Wv6cIJsK++62PHOnXq9CG2BjyZnDx5wuzCQGYXZjK7MJPZhZnMLsxlfmGmkydPXPKaSz4ucO99\nb/WL1X+tfqX6h9WPVW9aa32sem512977TPX2LgSxO6t37b1PVx+qjq+1Pl69tXrH47obAAAAAAAA\nqI6dP3/+qPfwaM6r6jCP/yMGZjK7MJPZhZnMLsxkdmEu8wsznTx54pKPKLzkCS4AAAAAAAB4MhG4\nAAAAAAAAGEXgAgAAAAAAYBSBCwAAAAAAgFEELgAAAAAAAEYRuAAAAAAAABhF4AIAAAAAAGAUgQsA\nAAAAAIBRBC4AAAAAAABGEbgAAAAAAAAYReACAAAAAABgFIELAAAAAACAUQQuAAAAAAAARhG4AAAA\nAAAAGEXgAgAAAAAAYBSBCwAAAAAAgFEELgAAAAAAAEYRuAAAAAAAABjl+FFv4NHc+MP/srNnHzrq\nbcCfcfbsn/YPvu/6rn/Rdx71VgAAAAAA4Ir2pAxc93/DX6tvOOpdwJ919k+/1P33P3DU2wAAAAAA\ngCueRxQCAAAAAAAwisAFAAAAAADAKAIXAAAAAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAA\nwCgCFwAAAAAAAKMIXAAAAAAAAIwicAEAAAAAADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACM\nInABAAAAAAAwisAFAAAAAADAKAIXAAAAAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgC\nFwAAAAAAAKMIXAAAAAAAAIwicAEAAAAAADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInAB\nAAAAAAAwisAFAAAAAADAKAIXAAAAAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAA\nAAAAAKMIXAAAAAAAAIwicAEAAAAAADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAA\nAAAwisAFAAAAAADAKAIXAAAAAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAA\nAKMIXAAAAAAAAIwicAEAAAAAADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAw\nisAFAAAAAADAKAIXAAAAAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMI\nXAAAAAAAAIwicAEAAAAAADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAF\nAAAAAADAKAIXAAAAAAAAowhcAAAAAAAAjHL8UhestW6ofqH6nepY9dvVv6g+0IVA9rnqjXvvB9da\nN1Y3Vw9Vt+y9b11rHa/eX72wOlu9ee/9mct/KwAAAAAAAFwJDnuC67/svV+x93753vvm6t3VT+29\nb6g+Xd201npm9c7qFdXLq7ettZ5TvaH64t77pdV7qvde9rsAAAAAAADginHYwHXsoq9fVt1x8PqO\n6lXVi6p79t4P7L3PVJ+oXlK9svrwwbV3Vy/+WjYMAAAAAADAle2wgevb1lq/tNb61bXWd1XP3Hs/\nePDefdULqudVpx7xmVMXr++9z1fnDh5bCAAAAAAAAI/ZYQLX/6jetff+e9UPVj/Tn/3dXRef7rrU\n+mGjGgAAAAAAAPw5lzxJtfe+t/qFg9e/v9b6o+o71lpX772/XF1Xfba6twsnth52XfXJg/XnV596\n+OTW3vvsZb0L+Dq59tpndPLkiaPexpOanw/MZHZhJrMLM5ldmMnswlzmF56aLhm41lpvqF6w9/6J\ntdbzu/DIwZ+tXld9sHpt9ZHqnup9a61rq3PV9dXN1bOr11d3Va+pPvoE3Ad8Xdx//5lOnTp91Nt4\n0jp58oSfDwxkdmEmswszmV2YyezCXOYXZjpMmD7M4wJ/ubphrfWr1Yert1T/pHrTWutj1XOr2/be\nZ6q3V3ce/HnX3vt09aHq+Frr49Vbq3c8jnsBAAAAAACA6nCPKHygCyevLvbqR7n29ur2i9bOVTc9\n3g0CAAAAAADAIx3mBBcAAAAAAAA8aQhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAA\nAKMIXAAAAAAAAIwicAEAAAAAADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAw\nisAFAAAAAADAKAIXAAAAAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMI\nXAAAAAAAAIwicAEAAAAAADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAF\nAAAAAADAKAIXAAAAAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMIXAAA\nAAAAAIwicAEAAAAAADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAFAAAA\nAADAKAIXAAAAAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMIXAAAAAAA\nAIwicAEAAAAAADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAFAAAAAADA\nKAIXAAAAAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMIXAAAAAAAAIwi\ncAEAAAAAADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAFAAAAAADAKAIX\nAAAAAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMIXAAAAAAAAIwicAEA\nAAAAADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAFAAAAAADAKAIXAAAA\nAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMIXAAAAAAAAIwicAEAAAAA\nADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAFAAAAAADAKMcPc9Fa6xnV\n71Tvrv5z9YEuxLHPVW/cez+41rqxurl6qLpl733rWut49f7qhdXZ6s17789c7psAAAAAAADgynHY\nE1zvrP73wet3Vz+1976h+nR101rrmQfXvKJ6efW2tdZzqjdUX9x7v7R6T/Xey7l5AAAAAAAArjyX\nDFxrrVX99epXqmPVDdUdB2/fUb2qelF1z977gb33meoT1UuqV1YfPrj27urFl3X3AAAAAAAAXHEO\nc4LrJ6p/1IW4VXXN3vvBg9f3VS+onledesRnTl28vvc+X507eGwhAAAAAAAAPC5fNXCttd5Y/fre\n+w++wiXHHuP6YR+JCAAAAAAAAI/qUqep/m71V9Za31tdV/1p9cBa6+q995cP1j5b3duFE1sPu676\n5MH686tPPXxya+999vLeAnz9XHvtMzp58sRRb+NJzc8HZjK7MJPZhZnMLsxkdmEu8wtPTV81cO29\nf+Dh12utH60+U11fva76YPXa6iPVPdX71lrXVucOrrm5enb1+uqu6jXVRy/7HcDX0f33n+nUqdNH\nvY0nrZMnT/j5wEBmF2YyuzCT2YWZzC7MZX5hpsOE6cfyyMCHHzv4Y9Wb1lofq55b3bb3PlO9vbrz\n4M+79t6nqw9Vx9daH6/eWr3jMfx9AAAAAAAA8Odc6hGF/9/e+58+4stXP8r7t1e3X7R2rrrpce8O\nAAAAAAAALvJYTnABAAAAAADAkRO4AAAAAAAAGEXgAgAAAAAAYBSBCwAAAAAAgFEELgAAAAAAAEYR\nuAAAAAAAABhF4AIAAAAAAGAUgQsAAAAAAIBRBC4AAAAAAABGEbgAAAAAAAAYReACAAAAAABgFIEL\nAAAAAACAUQQuAAAAAAAARhG4AAAAAAAAGEXgAgAAAAAAYBSBCwAAAAAAgFEELgAAAAAAAEYRuAAA\nAAAAABhF4AIAAAAAAGAUgQsAAAAAAIBRBC4AAAAAAABGEbgAAAAAAAAYReACAAAAAABgFIELAAAA\nAACAUQQuAAAAAAAARhG4AAAAAAAAGEXgAgAAAAAAYBSBCwAAAAAAgFEELgAAAAAAAEYRuAAAAAAA\nABhF4AIAAAAAAGAUgQsAAAAAAIBRBC4AAAAAAABGEbgAAAAAAAAYReACAAAAAABgFIELAAAAAACA\nUQQuAAAAAAAARhG4AAAAAAAAGEXgAgAAAAAAYBSBCwAAAAAAgFEELgAAAAAAAEYRuAAAAAAAABhF\n4AIAAAAAAGAUgQsAAAAAAIBRBC4AAAAAAABGEbgAAAAAAAAYReACAAAAAABgFIELAAAAAACAUQQu\nAAAAAAAARhG4AAAAAAAAGEXgAgAAAAAAYBSBCwAAAAAAgFEELgAAAAAAAEYRuAAAAAAAABhF4AIA\nAAAAAGAUgQsAAAAAAIBRBC4AAAAAAABGEbgAAAAAAAAYReACAAAAAABgFIELAAAAAACAUQQuAAAA\nAAAARhG4AAAAAAAAGEXgAgAAAAAAYBSBCwAAAAAAgFEELgAAAAAAAEYRuAAAAAAAABhF4AIAAAAA\nAGAUgQsAAAAAAIBRBC4AAAAAAABGEbgAAAAAAAAYReACAAAAAABgFIELAAAAAACAUQQuAAAAAAAA\nRhG4AAAAAAAAGEXgAgAAAAAAYBSBCwAAAAAAgFEELgAAAAAAAEYRuAAAAAAAABhF4AIAAAAAAGAU\ngQsAAAAAAIBRBC4AAAAAAABGEbgAAAAAAAAYReACAAAAAABgFIELAAAAAACAUY5f6oK11l+o3l89\nr7q6+mfVb1Uf6EIg+1z1xr33g2utG6ubq4eqW/bet661jh98/oXV2erNe+/PXPY7AQAAAAAA4Ipw\nmBNc31v9xt77ZdX3Vz9Zvbv66b33DdWnq5vWWs+s3lm9onp59ba11nOqN1Rf3Hu/tHpP9d7LfhcA\nAAAAAABcMS55gmvv/fOP+PIvV/+ruqF6y8HaHdWPVL9b3bP3fqBqrfWJ6iXVK6vbDq69u7r1suwc\nAAAAAACAK9KhfwfXWuvXqn9Xva26Zu/94MFb91Uv6MIjDE894iOnLl7fe5+vzh08thAAAAAAAAAe\ns0MHrr33i6vXVB+sjj3irWOP/omvuH7ovxMAAAAAAAAudsmTVGutb6/u23v/4d77t9daV1Wn11pX\n772/XF1Xfba6twsnth52XfXJg/XnV596+OTW3vvsZb4P+Lq49tpndPLkiaPexpOanw/MZHZhJrML\nM5ldmMnswlzmF56aDvOowL9dvbB621rredWzqv9Uva4Lp7leW32kuqd631rr2upcdX11c/Xs6vXV\nXV04AfbRy3wP8HVz//1nOnXq9FFv40nr5MkTfj4wkNmFmcwuzGR2YSazC3OZX5jpMGH6MI8L/LfV\nX1xr/Wp1R/XW6seqN621PlY9t7pt732ment158Gfd+29T1cfqo6vtT5+8Nl3PI57AQAAAAAAgOoQ\nJ7gOwtWNj/LWqx/l2tur2y9aO1fd9Hg3CAAAAAAAAI90mBNcAAAAAAAA8KQhcAEAAAAAADCKwAUA\nAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAFAAAAAADAKAIXAAAAAAAAowhcAAAA\nAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMIXAAAAAAAAIwicAEAAAAAADCKwAUAAAAA\nAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAFAAAAAADAKAIXAAAAAAAAowhcAAAAAAAA\njCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMIXAAAAAAAAIwicAEAAAAAADCKwAUAAAAAAMAo\nAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAFAAAAAADAKAIXAAAAAAAAowhcAAAAAAAAjCJw\nAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMIXAAAAAAAAIxy/Kg3AFMcO/a0PvZr/60//KMvHPVW\nnrSuuebq/viPv3zU27gifetfekHf9bKXHvU2AAAAAAC+LgQuOKSrnn51n3/6i/r8Z496J092J456\nA1eke7/waYELAAAAALhieEQhAAAAAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAA\nAAAAAKMIXAAAAAAAAIwicAEAAAAAADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAA\nAAAwisAFAAAAAADAKAIXAAAAAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAA\nAKMIXAAAAAAAAIwicAEAAAAAADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAw\nisAFAAAAAADAKAIXAAAAAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMI\nXAAAAAAAAIwicAEAAAAAADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAF\nAAAAAADAKAIXAAAAAAAAowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMIXAAA\nAAAAAIwicAEAAAAAADCKwAUAAAAAAMAoAhcAAAAAAACjHD/MRWutH69eUl1Vvbf6jeoDXQhkn6ve\nuPd+cK11Y3Vz9VB1y9771rXW8er91Qurs9Wb996fucz3AQAAAAAAwBXikie41lovq75t73199T3V\nv6reXf303vuG6tPVTWutZ1bvrF5Rvbx621rrOdUbqi/uvV9avacLgQwAAAAAAAAel8M8ovBj1esP\nXv+f6prqhuqXD9buqF5Vvai6Z+/9wN77TPWJLpz6emX14YNr765efHm2DgAAAAAAwJXokoFr731+\n7/2lgy9/qPqV6pq994MHa/dVL6ieV516xEdPXby+9z5fnTt4bCEAAAAAAAA8ZocOTWut76tuql5d\n/d4j3jr2FT7yldYPc2oMAAAAAAAAHtWhAtda67urd1Tfvfc+vdY6vda6eu/95eq66rPVvV04sfWw\n66pPHqw/v/rUwye39t5nL+M9AFzxrn7G8U6ePHHU22Aw/35gJrMLM5ldmMnswlzmF56aLhm41lrX\nVj9evXLv/X8Plu+uXlv93MF/P1LdU73v4Ppz1fXVzdWzu/A7vO6qXlN99DLfA8AV78tnznbq1Omj\n3gZDnTx5wr8fGMjswkxmF2YyuzCX+YWZDhOmD3OC6/urb6p+fq11rDpfvan6mbXWW6o/qG7bez+0\n1np7dWcXAte7Dk57fah61Vrr49WZ6gcfz80AAAAAAABAHSJw7b1vqW55lLde/SjX3l7dftHauS78\n7i4AAAAAAAD4mj3tqDcAAAAAAAAAj4XABQAAAAAAwCgCFwAAAAAAAKMIXAAAAAAAAIwicAEAAAAA\nADCKwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAFAAAAAADAKAIXAAAAAAAA\nowhcAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMIXAAAAAAAAIwicAEAAAAAADCK\nwAUAAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAFAAAAAADAKAIXAAAAAAAAowhc\nAAAAAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMIXAAAAAAAAIwicAEAAAAAADCKwAUA\nAAAAAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAFAAAAAADAKAIXAAAAAAAAowhcAAAA\nAAAAjCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMIXAAAAAAAAIwicAEAAAAAADCKwAUAAAAA\nAMAoAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAFAAAAAADAKAIXAAAAAAAAowhcAAAAAAAA\njCJwAQAAAAAAMIrABQAAAAAAwCgCFwAAAAAAAKMIXAAAAAAAAIwicAEAAAAAADCKwAUAAAAAAMAo\nAhcAAAAAAACjCFwAAAAAAACMInABAAAAAAAwisAFAAAAAADAKAIXAAAAAAAAowhcAAAAAAAAjCJw\nAQAAAAAAMIrABQAAAAAAwCjHj3oDj+ZLp79w1FvgCfINzzjRVU+/+qi3AQAAAAAADPakDFzHvvXO\no94CT5BTv/VNPf9bvvuotwEAAAAAAAz2pAxcz/2r33jUW+AJcv/vXHXUWwAAAAAAAIbzO7gAAAAA\nAAAYReACAAAAAABgFIELAAAAAACAUQQuAAAAAAAARhG4AAAAAAAAGEXgAgAAAAAAYBSBCwAAAAAA\ngFEELgAAAAAAAEYRuAAAAAAAABhF4AIAAAAAAGAUgQsAAAAAAIBRBC4AAAAAAABGEbgAAAAAAAAY\n5fhRbwCAr93n7/uj/v0vfviot8FQJ571jE4/cOYJ+d5XPe1Yr/2+7+2qq656Qr4/AAAAAFcmgQvg\nKeBLz/7O7vq9o94Fs139hHzXM1/47/2dV/9Jz3rWiSfk+wMAAABwZfKIQgAAAAAAAEYRuAAAAAAA\nABhF4AIAAAAAAGAUgQsAAAAAAIBRjh/morXW36h+qfrJvfe/WWt9S/WBLgSyz1Vv3Hs/uNa6sbq5\neqi6Ze9961rrePX+6oXV2erNe+/PXPY7AQAAAAAA4IpwyRNca61nVv+6uvsRy++ufmrvfUP16eqm\ng+veWb2ienn1trXWc6o3VF/ce7+0ek/13st7CwAAAAAAAFxJDvOIwjPV93ThpNbDXlbdcfD6jupV\n1Yuqe/beD+y9z1SfqF5SvbL68MG1d1cv/tq3DQAAAAAAwJXqkoFr731u7/3li5av2Xs/ePD6vuoF\n1fOqU4+45tTF63vv89W5g8cWAgAAAAAAwGN2mBNcl3LsMa5fjr8TAAAAAACAK9TjPUl1eq119cHJ\nruuqz1b3duHE1sOuqz55sP786lMPn9zae599/FsGAKZ42rFjffM3n+jEiRNHvRV4Sjp50mzBRGYX\nZjK7MJf5haemxxu47q5eW/3cwX8/Ut1TvW+tdW11rrq+url6dvX66q7qNdVHv8Y9AwBDnDt/vi98\n4XRnzhz1TuCp5+TJE506dfqotwE8RmYXZjK7MJf5hZkOE6YvGbjWWt9e/UT1wurBtdbrqhur29Za\nb6n+oLpt7/3QWuvt1Z1dCFzv2nufXmt9qHrVWuvj1ZnqBx/n/QAAAAAAAMClA9fe+zerlz/KW69+\nlGtvr26/aO1cddPj3SAAAAAAAAA80tOOegMAAAAAAADwWAhcAAAAAAAAjCJwAQAAwP9r7+5iLC8L\nO47/5m0XZwEXkJXdItBI8yCpF6JpCDRoSl+DCKmtGhvxJRpNNb5cmNoUaVVqDK1eVLQ2jYsx0bRg\nI9WqKViL2iKtxrYriTxCSsDdRSzKzm539mV25vTijK2x3Z3Z7Z55zjP7+VydmZz9n9/N2Zkz33P+\nfwAAoCsCFwAAAAAAAF0RuAAAAAAAAOiKwAUAAAAAAEBXBC4AAAAAAAC6InABAAAAAADQFYELAAAA\nAACArghcAAAAAAAAdEXgAgAAAAAAoCsCFwAAAAAAAF0RuAAAAAAAAOiKwAUAAAAAAEBXBC4AAAAA\nAAC6InABAAAAAADQFYELAAAAAACArghcAAAAAAAAdEXgAgAAAAAAoCsCFwAAAAAAAF0RuAAAAAAA\nAOiKwAUAAAAAAEBXBC4AAAAAAAC6InABAAAAAADQFYELAAAAAACArghcAAAAAAAAdEXgAgAAAAAA\noCsCFwAAAAAAAF0RuAAAAAAAAOiKwAUAAAAAAEBXBC4AAAAAAAC6InABAAAAAADQFYELAAAAAACA\nrghcAAAAAAAAdEXgAgAAAAAAoCsCFwAAAAAAAF0RuAAAAAAAAOiKwAUAAAAAAEBXBC4AAAAAAAC6\nInABAAAAAADQFYELAAAAAACArghcAAAAAAAAdEXgAgAAAAAAoCsCFwAAAAAAAF0RuAAAAAAAAOiK\nwAUAAADKaBa4AAAJMklEQVQAAEBXBC4AAAAAAAC6InABAAAAAADQlenWAwAAoIU3vePmLExubj3j\nhE1NT2bxyFLrGYzAzNKe3Pq+G1vPAAAAGGsCFwAAp6TBhrOzMHtJ6xknbKH1AEZmev6B1hMAAADG\nnlMUAgAAAAAA0BWBCwAAAAAAgK4IXAAAAAAAAHRF4AIAAAAAAKArAhcAAAAAAABdmW49AFgf5n7w\nnRye/HbrGYzIxMK5edqWK1rPAAAAAABIInABJ8nB+e/lohdvbD2DEXn0C3taTwAAAAAA+G9OUQgA\nAAAAAEBXBC4AAAAAAAC6InABAAAAAADQFYELAAAAAACArghcAAAAAAAAdEXgAgAAAAAAoCsCFwAA\nAAAAAF0RuAAAAAAAAOiKwAUAAAAAAEBXBC4AAAAAAAC6InABAAAAAADQFYELAAAAAACArghcAAAA\nAAAAdEXgAgAAAAAAoCvTrQcAAADwP+YPLeXD2/+i9QxGZHZ2Q+bnD7eecUKWlhZz7S9flQsveEbr\nKQAAIHABAACMk4mzLs03vt96BfxvRw4fyMX1QYELAICx4BSFAAAAAAAAdMUnuADgFHZo/sk8uffv\nMzWzYSTHXzw0lw9+am9mZmZGcnyObcumc3PD9Te0ngEAAABw0glcAHAKO3xgb879uYWcft7siB7h\nnOzMD0d0bFay95F9rScAAAAAjIRTFAIAAAAAANAVgQsAAAAAAICuCFwAAAAAAAB0ReACAAAAAACg\nK9OtBwAAAMdnsLSYXd+9IxvPnG09hRE4sj/Zev6LWs8AAAAYawIXAAB0ZjAYZNNPH8q2521uPYUR\n2PnlJ1tPAAAAGHtOUQgAAAAAAEBXfIILAAAA1sgPH//XLC4caD3jhCwuLuTrO2ZycHGu9ZSxNDM1\nnWt+6drWM05Zf/TBj+bRJw6N7PjT01M5cmRxZMdn/do/93hufe/vZHbWqaUBTrY1CVyllA8kuTzJ\nUpK31lq/sRaPCwAAAONk6cz7c/6VZ7WeccJ2J9mdf2g9YyzN379H4GrowJGp7H/KJaN9kJnRHp71\n6eD+ZGlJHAUYhZEHrlLKVUkurrVeUUq5JMn2JFeM+nEBAABg3ExMTGRyeqr1DEZgcspVIAAA1tJa\n/PZ1dZI7k6TW+kCSzaWU09fgcQEAAAAAAFiH1uIUhecl+fFTEj6x/L2H1uCxAQAAAEZv42T+9PaP\ntF5xyqq7dmRu8YGRHX/j9NY89Zxnj+z4rF/Tp2/Lm99zWyYmJlpPOWVNTU1mcXGp9YyxdPmzzs3r\nXvmy1jPghK3JNbh+wor/m+/86s5ksBZTWGsPfeVbeWz2vtYzGIF9c49lestzPHfXqYe/+u3snvpm\n6xmMwMH5uWw7tDWzZ+8byfGXFg9nanKQyQmn7Gnhvnu+ky/++d+0njG25v7zYE47Y0vrGSdkMFjK\nac8YZGl+ofUURmDXPz+cnXtvbj2DEZncfDDJRa1nnKClLC4cyIaZDa2HjK27H/u71hNOWYc3HsnU\nzJ6RHX9+566cMbKjs55NJhmmFX8waWYiAtdR3PlXn8iOb7q2JuPnmc+8OH/4nneueL+1CFy7M/zE\n1o9sS/LYsf7Bvbd+1lsa1qsPtR4AAAAAAEByY+sB8P+yFm+nvivJbyRJKeWyJLtqrfvX4HEBAAAA\nAABYhyYGg9F/PLaU8t4kz0+ymOSNtdZvjfxBAQAAAAAAWJfWJHABAAAAAADAyeKK7wAAAAAAAHRF\n4AIAAAAAAKArAhcAAAAAAABdmW494CeVUn42yZ1JPlBr/XDrPcDqlFJuSfLzSaaSvK/W+unGk4AV\nlFKekuRjSZ6eZGOSm2utn2s6Cli1UsppSe5P8u5a68db7wGOrZTy/CR3ZPi8nUiyo9b6lrargNUo\npfxWkrcnWUhyU631C40nASsopbwmySuSDDL8ufvcWuuZbVcBKymlbEry8SRnJdmQ4evdu452/7EK\nXKWU2SR/kuSLrbcAq1dKeUGSS2utV5RSzk7yL0kELhh/1yb5eq31j0spFyS5O4nABf14Z5IftB4B\nHJd7aq0vaT0CWL3l17g3JXlOkjOSvCuJwAVjrta6Pcn2JCmlXJXkN9suAlbpVUkeqLX+Xilla5Iv\nJXnW0e48VoErycEkv5bkHa2HAMfly0n+afn2niSzpZSJWuug4SZgBbXW23/sywuSfLfVFuD4lFJK\nkksiSkNvJloPAI7bLya5u9Y6n2Q+yRsa7wGO301JXt56BLAqTyR59vLts5P8x7HuPFaBq9a6lOTQ\n8PU60IvlkHVg+cvXJvm8uAX9KKX8Y5KfSvLC1luAVXt/kjdm+O42oB+XllLuzPDF+rtrrc5eAuPv\noiSbSil/nWRzknfVWr/UdhKwWqWU5yV5tNb6/dZbgJXVWv+ylPKqUsqDGf7cveZY959cm1nAqaCU\ncl2SVyd5U+stwOrVWq9Mcl2ST7TeAqyslPKKJPfWWh9Z/pZPhEAfHkzyB7XW6zOM0x8tpYzVm06B\n/9NEhlH6+gxf797Wdg5wnF6b4bWngQ4sX/fykVrrzyS5OsmHjnV/gQs4KUopv5Lkd5P8aq11X+s9\nwMpKKZeVUs5PklrrvyWZLqU8rfEsYGXXJLmulPK1DF+w31hK+YXGm4AV1Fp311rvWL7970m+l+En\nqIHx9niGbywZLD939/mdGbrygiT3th4BrNqVSf42SWqtO5JsK6Uc9U2d4/xuMe9EhU6UUs5MckuS\nq2utc633AKt2VZILk7ytlPL0JJtqrU803gSsoNb6sh/dLqX8fpKHnSoJxl8p5eVJttZa319KOS/J\nliS7Gs8CVnZXkttKKbdk+EkuvzNDJ0opW5Psq7Ueab0FWLWHklye5NOllAszfA4f9VI4YxW4SimX\nZXg9gQuTLJRSXpzk12ute9ouA1bw0iTnJLl9uagPktxQa93Zdhawgo9keHqkryQ5LclvN94DAOvZ\nZ5J8cvm03jNJ3uAPbjD+aq27SymfSnJfhq91nZIf+rE1iWtvQV/+LMn2Uso9SaaSvP5Yd54YDI4a\nvwAAAAAAAGDsuAYXAAAAAAAAXRG4AAAAAAAA6IrABQAAAAAAQFcELgAAAAAAALoicAEAAAAAANAV\ngQsAAAAAAICuCFwAAAAAAAB0ReACAAAAAACgK/8FlwsPcWsielgAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7fa9a3e22358>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Create a distribution graph to identify the number of pure nuclear interacting or pure cytosol interacting RNAs  \n",
    "distrib_nucleus = []\n",
    "distrib_cytosol = []\n",
    "for gene_id, val in gene_loc_dict.items():\n",
    "    if not val[\"nucleus\"]:\n",
    "        distrib_cytosol.append(val[\"cytosol\"])\n",
    "    elif not val[\"cytosol\"]:\n",
    "        distrib_nucleus.append(val[\"nucleus\"])\n",
    "\n",
    "distrib_nucleus.sort()\n",
    "distrib_cytosol.sort()\n",
    "p= pl.hist(distrib_nucleus,align=\"mid\", histtype=\"stepfilled\")\n",
    "p= pl.hist(distrib_cytosol,align=\"mid\", histtype=\"stepfilled\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAABr4AAAJSCAYAAACV2LUOAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAIABJREFUeJzs3X+s3fV93/HXBWfpHJsfRhdMvcnQKPtUWpQ/okmZHEjK\nr0TRFrTyo5ugUQP/ZBFDjGhIdBIJYxtCm2AJWZNlRBSaphLSBtlQlQwsaAYLgz/4I0xl73VJnUox\nnW963cpeamJ87/64x9ONa+zLsZvLed/HQ7J07ud8zjmfr3Q/5/7x9OecueXl5QAAAAAAAMCsO2O9\nFwAAAAAAAACng/AFAAAAAABAC8IXAAAAAAAALQhfAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAub\n1jJpjPHeJN9I8kBVfWnV+EeTfLOqzpj8fGOS25IcSfJQVT08xtiU5JEkO5O8keSmqtozxnhfki8n\nWUry3aq65fRdFgAAAAAAABvNSU98jTE2J3kwye5jxt+Z5M4ke1fNuyvJ5UkuS3L7GOOcJDck2V9V\nlya5N8l9k6f4fJJbJ+PnTCIaAAAAAAAATGUtH3V4KMnHkrx2zPg/TfJvk/xk8vMHkrxUVQer6lCS\n55NckuSKJE9M5uxOsmuM8Y4kF1fVy5PxJ5NcOfVVAAAAAAAAsOGdNHxV1VJVvb56bIzxN5K8r6r+\n46rh7UkWVv28kOTCJBccHa+q5STLk7mLq+bum8wFAAAAAACAqazpO76O44Ekt05uz73JnBONLx9z\n/5vNBQAAAAAAgDV5y+FrjPHzSUaSr48x5pJcOMZ4Nsnnknx81dQdSV7IyneAbU/yyhhjU1Yi12tJ\nzjtm7t4Tve7y8vLy3Jw+BgAAAAAAsAGtKRK91fA1V1V7k7zn6MAY4w+r6rIxxs8l+eoY46wkS0l2\nJbktydlJrk/ydJKrkzxbVUfGGK+OMXZV1XeSXJPkwRO+8NxcFhYOvMXlAuttfn6rvQszyv6F2WTv\nwmyyd2E22bswm+xdmE3z81vXNO+k4WuM8f4k9yfZmeTwGOPaJNdU1Z9OpiwnSVUdGmPcmeSprISv\nu6vqwBjjsSRXjTGeS3IoyScnj7s9yVcmp8ZerKpn1npxAAAAAAAAcKy55eXl9V7DWi2r8DB7/A8a\nmF32L8wmexdmk70Ls8nehdlk78Jsmp/fuqaPOjzjL3shAAAAAAAA8LMgfAEAAAAAANCC8AUAAAAA\nAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IXAAAAAAAALQhfAAAAAAAA\ntCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEAAAAAANCC8AUAAAAAAEAL\nwhcAAAAAAAAtCF8AAAAAAAC0sGm9FwAAAMCJHTlyJHv2fH+9l8Ffov37t2Rx8eB6L2NqF130Cznz\nzDPXexkAACB8AQAAvN3t2fP93Pav/3M2n33+ei8F/oIf/9m+fOGOq/Pud79nvZcCAADCFwAAwCzY\nfPb52XLujvVeBgAAwNua7/gCAAAAAACgBeELAAAAAACAFoQvAAAAAAAAWhC+AAAAAAAAaEH4AgAA\nAAAAoAXhCwAAAAAAgBaELwAAAAAAAFoQvgAAAAAAAGhB+AIAAAAAAKAF4QsAAAAAAIAWhC8AAAAA\nAABaEL4AAAAAAABoQfgCAAAAAACgBeELAAAAAACAFoQvAAAAAAAAWhC+AAAAAAAAaEH4AgAAAAAA\noAXhCwAAAAAAgBaELwAAAAAAAFoQvgAAAAAAAGhB+AIAAAAAAKAF4QsAAAAAAIAWhC8AAAAAAABa\nEL4AAAAAAABoQfgCAAAAAACgBeELAAAAAACAFoQvAAAAAAAAWhC+AAAAAAAAaEH4AgAAAAAAoAXh\nCwAAAAAAgBaELwAAAAAAAFoQvgAAAAAAAGhB+AIAAAAAAKAF4QsAAAAAAIAWhC8AAAAAAABaEL4A\nAAAAAABoQfgCAAAAAACgBeELAAAAAACAFoQvAAAAAAAAWhC+AAAAAAAAaEH4AgAAAAAAoAXhCwAA\nAAAAgBaELwAAAAAAAFoQvgAAAAAAAGhB+AIAAAAAAKAF4QsAAAAAAIAWhC8AAAAAAABaEL4AAAAA\nAABoQfgCAAAAAACgBeELAAAAAACAFoQvAAAAAAAAWti0lkljjPcm+UaSB6rqS2OMv57k4STvSPKT\nJL9aVfvGGDcmuS3JkSQPVdXDY4xNSR5JsjPJG0luqqo9Y4z3JflykqUk362qW07ztQEAAAAAALCB\nnPTE1xhjc5IHk+xeNfzPk/y7qvqlrASxz0zm3ZXk8iSXJbl9jHFOkhuS7K+qS5Pcm+S+yXN8Psmt\nk/FzxhgfPT2XBAAAAAAAwEa0lo86PJTkY0leWzX26SSPT24vJDkvyQeSvFRVB6vqUJLnk1yS5Iok\nT0zm7k6ya4zxjiQXV9XLk/Enk1x5KhcCAAAAAADAxnbS8FVVS1X1+jFjf15Vy2OMM5LckuR3kmzP\nSgQ7aiHJhUkuODpeVctJlidzF1fN3TeZCwAAAAAAAFNZy4mv45pEr68l2V1Vzx5nytybPHQuK/Fr\nbg1zAQAAAAAAYE02ncJjfzNJVdW/mPy8Nz99amtHkhcm49uTvDLG2JSVyPVaVj4ecfXcvSd7wfn5\nraewXGC92Lswu+xfmE32bj/7929Z7yXACW3btsV7DxuW332YTfYu9DVV+Bpj3Jjk9aq6Z9Xwi0ke\nGmOclWQpya4ktyU5O8n1SZ5OcnWSZ6vqyBjj1THGrqr6TpJrkjx4stddWDgwzXKBdTQ/v9XehRll\n/8Jssnd7Wlw8uN5LgBNaXDzovYcNyd9dmE32LsymtQbrk4avMcb7k9yfZGeSw2OM65Kcn+TQGOPZ\nrHxs4e9X1T8aY9yZ5KmshK+7q+rAGOOxJFeNMZ5LcijJJydPfXuSr4wx5pK8WFXPvJULBAAAAAAA\ngNVOGr6q6uUkl63lyarq8SSPHzO2lOTm48x9NcmH1rZMAAAAAAAAOLEz1nsBAAAAAAAAcDoIXwAA\nAAAAALQgfAEAAAAAANCC8AUAAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAA\nAABAC8IXAAAAAAAALQhfAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAA\nALQgfAEAAAAAANCC8AUAAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABA\nC8IXAAAAAAAALQhfAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQg\nfAEAAAAAANCC8AUAAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IX\nAAAAAAAALQhfAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEA\nAAAAANCC8AUAAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IXAAAA\nAAAALQhfAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEAAAAA\nANCC8AUAAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IXAAAAAAAA\nLQhfAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEAAAAAANCC\n8AUAAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IXAAAAAAAALQhf\nAAAAAAAAtCB8AQAAAAAA0MKmtUwaY7w3yTeSPFBVXxpj/LUkX8tKOHstySeq6vAY48YktyU5kuSh\nqnp4jLEpySNJdiZ5I8lNVbVnjPG+JF9OspTku1V1y2m+NgAAAAAAADaQk574GmNsTvJgkt2rhu9J\n8sWq+nCS7yW5eTLvriSXJ7ksye1jjHOS3JBkf1VdmuTeJPdNnuPzSW6djJ8zxvjoabomAAAAAAAA\nNqC1fNThoSQfy8rJrqN+KcmTk9tPJrkqyQeSvFRVB6vqUJLnk1yS5IokT0zm7k6ya4zxjiQXV9XL\nq57jylO4DgAAAAAAADa4k4avqlqqqtePGX5XVR2e3N6X5MIkFyRZWDVn4djxqlpOspxke5LFVXOP\nPgcAAAAAAABMZU3f8XUSc1OMLx9z/5vN/Snz81vfwrKAtwt7F2aX/Quzyd7tZ//+Leu9BDihbdu2\neO9hw/K7D7PJ3oW+pg1fB8YY75ycBNuR5IdJ9uanT23tSPLCZHx7klfGGJuyErleS3LeMXP3nuxF\nFxYOTLlcYL3Mz2+1d2FG2b8wm+zdnhYXD673EuCEFhcPeu9hQ/J3F2aTvQuzaa3Bei3f8XU8u5Nc\nO7l9bZJvJXkpyd8aY5w1xtiSZFeS55I8neT6ydyrkzxbVUeSvDrG2DUZv2byHAAAAAAAADCVk574\nGmO8P8n9SXYmOTzGuC7JjUkeHWN8KskPkjxaVUfGGHcmeSrJUpK7q+rAGOOxJFeNMZ5LcijJJydP\nfXuSr4wx5pK8WFXPnOZrAwAAAAAAYAM5afiqqpeTXHacuz5ynLmPJ3n8mLGlJDcfZ+6rST605pUC\nAAAAAADACUz7UYcAAAAAAADwtiJ8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQg\nfAEAAAAAANCC8AUAAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IX\nAAAAAAAALQhfAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEA\nAAAAANCC8AUAAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IXAAAA\nAAAALQhfAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEAAAAA\nANCC8AUAAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IXAAAAAAAA\nLQhfAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEAAAAAANCC\n8AUAAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IXAAAAAAAALQhf\nAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEAAAAAANCC8AUA\nAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IXAAAAAAAALQhfAAAA\nAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEAAAAAANCC8AUAAAAA\nAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQwqZpHjTGeFeS30pybpK/kuSeJH+c5MtJlpJ8\nt6pumcy9I8l1k/F7quqbY4yzkvxOkrOTHEhyQ1X96SleCwAAAAAAABvYtCe+Ppnkf1bV5VmJWl9I\n8m+S3FpVlyY5Z4zx0THGRUl+JcmuJB9P8sAYYy7JP07y7GTuE0nuPKWrAAAAAAAAYMObNnz9KMl5\nk9vnJfmTJBdX1cuTsSeTXJXksiTfrKojVfWjJHuS/M0kV2QleB2de+WU6wAAAAAAAIAkU4avqnos\nyc4xxh8k+b0kdyTZv2rKviQXJrkgycJJxvcl2T7NOgAAAAAAAOCoqcLXGOPGJD+oqvckuTzJbx8z\nZe4tvN6bzQUAAAAAAIA12zTl4z6Y5L8kSVW9Msb4q8c8144kP0yyN8kvvsn49iQHJmN71/Ki8/Nb\np1wusJ7sXZhd9i/MJnu3n/37t6z3EuCEtm3b4r2HDcvvPswmexf6mjZ8/e8kfzvJE2OMnVkJWH84\nxvhgVf23JNckeTDJHyT5zBjjs0nOT/LzVfX7Y4ynk/xKkn+Z5Nok31rLiy4sHJhyucB6mZ/fau/C\njLJ/YTbZuz0tLh5c7yXACS0uHvTew4bk7y7MJnsXZtNag/W04esrSR4eY/xekjOTfCrJHyf592OM\nuSQvVtUzSTLGeCjJc0mWkvzDyeMfTPLbY4z/mpXvBvvVKdcBAAAAAAAASaYMX1X1f5P8/ePc9aHj\nzP2NJL9xnMf/8jSvDQAAAAAAAMdzxnovAAAAAAAAAE4H4QsAAAAAAIAWhC8AAAAAAABaEL4AAAAA\nAABoQfgCAAAAAACgBeELAAAAAACAFoQvAAAAAAAAWhC+AAAAAAAAaEH4AgAAAAAAoAXhCwAAAAAA\ngBaELwAAAAAAAFoQvgAAAAAAAGhB+AIAAAAAAKAF4QsAAAAAAIAWhC8AAAAAAABaEL4AAAAAAABo\nQfgCAAAAAACgBeELAAAAAACAFoQvAAAAAAAAWhC+AAAAAAAAaEH4AgAAAAAAoAXhCwAAAAAAgBaE\nLwAAAAAAAFoQvgAAAAAAAGhB+AIAAAAAAKAF4QsAAAAAAIAWhC8AAAAAAABaEL4AAAAAAABoQfgC\nAAAAAACgBeELAAAAAACAFoQvAAAAAAAAWhC+AAAAAAAAaEH4AgAAAAAAoAXhCwAAAAAAgBaELwAA\nAAAAAFoQvgAAAAAAAGhB+AIAAAAAAKAF4QsAAAAAAIAWhC8AAAAAAABaEL4AAAAAAABoQfgCAAAA\nAACgBeELAAAAAACAFoQvAAAAAAAAWhC+AAAAAAAAaEH4AgAAAAAAoAXhCwAAAAAAgBaELwAAAAAA\nAFoQvgAAAAAAAGhB+AIAAAAAAKAF4QsAAAAAAIAWhC8AAAAAAABaEL4AAAAAAABoQfgCAAAAAACg\nBeELAAAAAACAFoQvAAAAAAAAWhC+AAAAAAAAaEH4AgAAAAAAoAXhCwAAAAAAgBaELwAAAAAAAFoQ\nvgAAAAAAAGhB+AIAAAAAAKAF4QsAAAAAAIAWhC8AAAAAAABaEL4AAAAAAABoQfgCAAAAAACgBeEL\nAAAAAACAFoQvAAAAAAAAWhC+AAAAAAAAaEH4AgAAAAAAoAXhCwAAAAAAgBaELwAAAAAAAFrYNO0D\nxxg3JrkjyeEkn03ySpKvZSWmvZbkE1V1eDLvtiRHkjxUVQ+PMTYleSTJziRvJLmpqvacwnUAAAAA\nAACwwU114muMsS0rsWtXkr+b5O8luSfJF6vqw0m+l+TmMcbmJHcluTzJZUluH2Ock+SGJPur6tIk\n9ya571QvBAAAAAAAgI1t2hNfVyZ5uqp+nOTHST41xvh+kk9N7n8yyT9J8r+SvFRVB5NkjPF8kkuS\nXJHk0cnc3UkennIdAAAAAAAAkGT67/i6KMm7xhj/aYzx7THG5Uk2V9Xhyf37klyY5IIkC6set3Ds\neFUtJ1mafPwhAAAAAAAATGXa2DSXZFuSX85KBHt2Mrb6/jd73PFMG+AAAAAAAAAgyfTh6/8k+U5V\nLSX5/hjjQJLDY4x3VtXrSXYk+WGSvVk54XXUjiQvTMa3J3nl6EmvqnrjZC86P791yuUC68nehdll\n/8Jssnf72b9/y3ovAU5o27Yt3nvYsPzuw2yyd6GvacPXU0l+c4zxr7Jy8mtLkm8luS7J15NcO/n5\npSRfHWOclWQpya4ktyU5O8n1SZ5OcnVWToyd1MLCgSmXC6yX+fmt9i7MKPsXZpO929Pi4sH1XgKc\n0OLiQe89bEj+7sJssndhNq01WE/1EYNVtTfJf0jy35P8bpJbknwuya+NMb6d5Nwkj1bVoSR3ZiWU\nPZXk7qo6kOSxJJvGGM8l+XSSX59mHQAAAAAAAHDUtCe+UlUPJXnomOGPHGfe40keP2ZsKcnN0742\nAAAAAAAAHGuqE18AAAAAAADwdiN8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQg\nfAEAAAAAANCC8AUAAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IX\nAAAAAAAALQhfAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEA\nAAAAANCC8AUAAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IXAAAA\nAAAALQhfAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEAAAAA\nANCC8AUAAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IXAAAAAAAA\nLQhfAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEAAAAAANCC\n8AUAAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IXAAAAAAAALQhf\nAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEAAAAAANCC8AUA\nAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IXAAAAAAAALQhfAAAA\nAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEAAAAAANCC8AUAAAAA\nAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC5tO5cFjjJ9L8j+S3JPkmSRf\ny0pMey3JJ6rq8BjjxiS3JTmS5KGqeniMsSnJI0l2JnkjyU1VtedU1gIAAAAAAMDGdqonvu5K8ieT\n2/ck+WJVfTjJ95LcPMbYPJlzeZLLktw+xjgnyQ1J9lfVpUnuTXLfKa4DAAAAAACADW7q8DXGGEl+\nMcnvJplL8uEkT07ufjLJVUk+kOSlqjpYVYeSPJ/kkiRXJHliMnd3kg9Ouw4AAAAAAABITu3E1/1J\nPpOV6JUk76qqw5Pb+5JcmOSCJAurHrNw7HhVLSdZmnz8IQAAAAAAAExlqtg0xvhEku9U1Q9WDn79\nBXPHGzzB+JoC3Pz81rVMA95m7F2YXfYvzCZ7t5/9+7es9xLghLZt2+K9hw3L7z7MJnsX+pr2lNXf\nSXLxGOPjSXYk+UmSg2OMd1bV65OxHybZm5UTXkftSPLCZHx7kleOnvSqqjdO9qILCwemXC6wXubn\nt9q7MKPsX5hN9m5Pi4sH13sJcEKLiwe997Ah+bsLs8nehdm01mA9Vfiqqn9w9PYY47NJ9iTZleS6\nJF9Pcm2SbyV5KclXxxhnJVmazLktydlJrk/ydJKrkzw7zToAAAAAAADgqFP5jq+jjn584eeS/NoY\n49tJzk3yaFUdSnJnkqcm/+6uqgNJHkuyaYzxXJJPJ/n107AOAAAAAAAANrBpP+rw/6uqf7bqx48c\n5/7Hkzx+zNhSkptP9bUBAAAAAADgqNNx4gsAAAAAAADWnfAFAAAAAABAC8IXAAAAAAAALQhfAAAA\nAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEAAAAAANCC8AUAAAAA\nAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IXAAAAAAAALQhfAAAAAAAA\ntCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEAAAAAANCC8AUAAAAAAEAL\nwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IXAAAAAAAALQhfAAAAAAAAtCB8\nAQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0IXwAAAAAAALQgfAEAAAAAANCC8AUAAAAAAEALwhcA\nAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAFAAAAAABAC8IXAAAAAAAALQhfAAAAAAAAtCB8AQAA\nAAAA0MKm9V4AAAAAMLuWl5byR3/0g/VeBhzXRRf9Qs4888z1XgYA8DMkfAEAAABT+/MDC7n/sR9l\n89mvrfdS4Kf8+M/25Qt3XJ13v/s9670UAOBnSPgCAAAATsnms8/PlnN3rPcyAADAd3wBAAAAAADQ\ng/AFAAAAAABAC8IXAAAAAAAALQhfAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAAAC0I\nXwAAAAAAALQgfAEAAAAAANCC8AUAAAAAAEALwhcAAAAAAAAtCF8AAAAAAAC0IHwBAAAAAADQgvAF\nAAAAAABAC8IXAAAAAAAALQhfAAAAAAAAtCB8AQAAAAAA0ILwBQAAAAAAQAvCFwAAAAAA8P/au/9Y\nveq7DuDvUtSlXFZ+hIJCAtLND5C4RBKVMAZsOAhxwpKJMxKUjSUYMQEjJPoH+xWzIIQ5jFliuoFu\nCbHEjInZRmBxTidzc0t0LMI3Qr298iPQckvX0uLg3vrH8zS20NveXtp+73N4vZLmPufcc5/n3eQ5\n9zznvs/3e2AQju4dAAAAlou5ublMT2/oHeMN2bJlKrOz23vH4BCbmdnYOwIAAMBEUHwBAEfcEMoF\nhmlmZmPuXP8fWbV6Te8osJcXnnosJ552du8YAAAAy57iCwA44qanN+TGOx5QLrDs7C4Xpo4/tXcU\n2MuOrc/1jgAAADARFF8AQBerVq9RLrDsKBcAAABgsh3VOwAAAAAAAAAcCoovAAAAAAAABkHxBQAA\nAAAAwCAs+R5fVXV7kguSrExyW5J/S/LFjMq0Z5Nc01p7paquTnJjkrkk61prd1fV0Un+KsnpSV5N\n8qHW2vQb+H8AAAAAAADwJrekEV9VdXGSc1pr5ye5PMlnknwyyV+01i5K8mSSD1fVqiS3JnlPkncn\n+YOqOi7EZoxkAAALdElEQVTJbyXZ0lp7V5JPZVScAQAAAAAAwJItdarDbya5avz4xSTHJLkoyQPj\ndX+f5L1JfjnJd1tr21trLyf5VkajxC5Jcv94268neecScwAAAAAAAECSJRZfrbVdrbWd48Xrknwl\nyTGttVfG655P8tNJTk6yaY8f3fTa9a21XUnmx9MfAgAAAAAAwJK8obKpqq5M8uEklyZ5Yo9vrVjg\nRxZav9SRZwAAAAAAAJDkDRRfVXVZkj9OcllrbVtVbauqn2qt/W+SU5M8neSZjEZ47XZqkm+P15+S\n5NHdI71aa68e6DVPOunYpcYFOrLvwuQ6XPvvli1Th+V5AQBgTyecMHXYz0md88Jksu/CcC2p+Kqq\ntya5PcklrbWt49VfT/KBJPeOvz6Y5LtJPjfefj7J+UluTLI6o3uEPZzkiiTfWMzrbtq0bSlxgY5O\nOulY+y5MqMO5/87Obj8szwsAAHuand1+WM9JnfPCZLLvwmRabGG91BFfH0xyYpL7qmpFkl1JfifJ\n56vq+iQbk/x1a22uqv4oyUMZFV8fH48OW5/kvVX1z0leTnLtEnMAAAAAAABAkiUWX621dUnW7eNb\nl+5j2y8l+dJr1s1ndG8wAAAAAAAAOCSWfI8vAAAAAFiuds3PZ2Zm42F9jS1bpkzjzZKdccaZWbly\nZe8YAIOj+AIAAABgcHZu25Q712/OqtXP9o4Cr7Nj6/O565Yrsnbt23tHARgcxRcAAAAAg7Rq9ZpM\nHX9q7xgAwBF0VO8AAAAAAAAAcCgovgAAAAAAABgExRcAAAAAAACDoPgCAAAAAABgEBRfAAAAAAAA\nDILiCwAAAAAAgEFQfAEAAAAAADAIii8AAAAAAAAGQfEFAAAAAADAICi+AAAAAAAAGATFFwAAAAAA\nAIOg+AIAAAAAAGAQFF8AAAAAAAAMguILAAAAAACAQVB8AQAAAAAAMAiKLwAAAAAAAAZB8QUAAAAA\nAMAgKL4AAAAAAAAYBMUXAAAAAAAAg6D4AgAAAAAAYBAUXwAAAAAAAAyC4gsAAAAAAIBBUHwBAAAA\nAAAwCIovAAAAAAAABkHxBQAAAAAAwCAovgAAAAAAABgExRcAAAAAAACDoPgCAAAAAABgEI7uHQCA\nw2Nubi7T0xt6x2CCbdkyldnZ7YfluWdmNh6W5wUAAADgzU3xBTBQ09MbcuMdD2TV6jW9o8DrvPDU\nYznxtLN7xwAAAABgYBRfAAO2avWaTB1/au8Y8Do7tj7XOwIAAAAAA+QeXwAAAAAAAAyC4gsAAAAA\nAIBBUHwBAAAAAAAwCIovAAAAAAAABkHxBQAAAAAAwCAovgAAAAAAABgExRcAAAAAAACDoPgCAAAA\nAABgEBRfAAAAAAAADILiCwAAAAAAgEFQfAEAAAAAADAIii8AAAAAAAAGQfEFAAAAAADAICi+AAAA\nAAAAGATFFwAAAAAAAIOg+AIAAAAAAGAQFF8AAAAAAAAMguILAAAAAACAQVB8AQAAAAAAMAiKLwAA\nAAAAAAZB8QUAAAAAAMAgKL4AAAAAAAAYBMUXAAAAAAAAg6D4AgAAAAAAYBAUXwAAAAAAAAyC4gsA\nAAAAAIBBUHwBAAAAAAAwCIovAAAAAAAABkHxBQAAAAAAwCAovgAAAAAAABgExRcAAAAAAACDoPgC\nAAAAAABgEBRfAAAAAAAADILiCwAAAAAAgEE4uncAmHRzc3OZnt7QO8aytWXLVGZnt/eO8aY0M7Ox\ndwQAAAAAgCNK8QVv0PT0htx4xwNZtXpN7yiwlxeeeiwnnnZ27xgAAADAa+yan3fBakcu1N6/M844\nMytXruwdA5ZM8QWHwKrVazJ1/Km9Y8Bedmx9rncEAAAAYB92btuUO9dvzqrVz/aOAnvZsfX53HXL\nFVm79u29o8CSdS2+qurTSc5LMp/kptba93rmAQAAAAA4ElxIDXB4dCu+qurCJG9rrZ1fVWcluTvJ\n+Qttf/Otd2bLSyuOWD5YrO2zM8lRP9c7BgAAAAAAvOn1HPF1SZIvJ0lr7fGqOq6qplpr+5xc9blt\nyYsr1x7RgLAYP97xdDLVOwUAAAAAANCz+DolyZ5TG24er3uiTxwAAAAAAHjz2jU/n5mZjb1jwD6d\ndNK5i9qu6z2+XmO/8xjOvbQp86/sPFJZYNFe3bk5O+aO7R0DXmfnttkc4FcrdOP9yXLlvcly5b3J\ncub9yXLlvcly5v3JcjX7TMufrPvPvGXqhN5RYC8vb5/NDx7+7KK27Vl8PZPRCK/dfibJswttfO+6\n2xwJAAAAAAAAWNBRHV/7oSS/niRVdW6Sp1trL3XMAwAAAAAAwARbsWvXrm4vXlWfSnJRkrkkN7TW\nHu0WBgAAAAAAgInWtfgCAAAAAACAQ6XnVIcAAAAAAABwyCi+AAAAAAAAGATFFwAAAAAAAINwdO8A\nB6Oqbk5ydZIfJ/m91tr3O0cCFqmqTk7yWJL3t9b+qXceYP+qamWSzydZm2Rlkptba4/0TQXsT1V9\nOsl5SeaT3NRa+17nSMAiVdXtSS7I6Jh7W2vt/s6RgEWoqrck+WGST7bWvtA7D7A4VXV1kluSvJLk\no621r3WOBBxAVR2T5AtJjk/ykxkdex9aaPuJGfFVVeck+Y0k5ya5Psn7+iYCDtLtSZ7sHQJYtGuS\nbG+tvSvJR5L8Wec8wH5U1YVJ3tZaOz+jffbPO0cCFqmqLk5yznj/vTzJZ/omAg7CrUle6B0CWLyq\nOiHJR5Ocn9Hfl6/smwhYpGuTPN5ae0+Sq5Lctb+NJ2nE1/uS3Nda25Xk38f/gAlQVe9O8qMkj/bO\nAizaF5PcO368KckJHbMAB3ZJki8nSWvt8ao6rqqmWmvbO+cCDuybSb4zfvxiklVVtWJ87gssU1VV\nSc5K8pXeWYCD8itJHm6t7UiyI8nvds4DLM7mJD8/fnxCRn+rWtAkFV9nJJmrqq9llPsPW2s/6BsJ\nOJCq+omMrqS5Mgdo4oHlo7U2l2RuvHhT/r8EA5anU5LsObXh5vG6J/rEARZrXHDtHC9+JMlXlV4w\nEe5MckNGV6ADk+OMJMdU1d8lOS7JJ1pr/9A3EnAgrbX1VXVtVf1XRvvur+5v+2VZfFXVdRl94N/9\nYX9FkjVJHmytXV5V70zyuSS/1CkisA+v2XdXjL8+mGRda+1HowvisqJfQmBfFth3P9Zae7iqbkjy\nC0l+rWNE4OA53sKEqaork3woyaW9swD7V1XXJHmktbbReS5MnBUZjRZ5f5KfTfKNJKd3TQQc0Pje\nfBvH/dA7Mrov/S8utP2KXbsm40KyqvpYRnM4rh8vP9daO7lzLOAAqupbGd1PcEWStUmeT3JVa+2x\nrsGAAxoXYh9IcmVr7ZXeeYCFjT8rP9NaWzdefjLJO1prL/VNBixGVV2W5BNJLmutbe2dB9i/qvqb\njP5gPp/ktCQvJ7neqBFY/qrq2iQnt9b+dLz8wyQXt9Y2dw0G7FdVfTajaUrvHy8/neS0hWZKWJYj\nvhbwYEZzrq6vqrOS/E/nPMAitNYu2P24qu5Jco/SC5a/qjozyfVJLlR6wUR4KMnHk6yrqnOTPK30\ngslQVW9NcnuSS5ReMBlaa7+5+/H44pP/VnrBxHgoyT1VdXtGI7+OUXrBRHgiyXlJ7q+q05Ns29/0\n4BNTfLXWvlNVl1fVIxlNwXRD70zAQZuMIaZAklyX0UnAV6tq9/SHl7bWXu0bC9iX1tq3q+r7VfUv\nGd2fz2dlmBwfTHJikvv2OOb+dmvtqb6xAGB4WmvPVNXfJvnXjI65v985ErA4f5nk7qr6xyQrM7pY\ne0ETM9UhAAAAAAAA7M9RvQMAAAAAAADAoaD4AgAAAAAAYBAUXwAAAAAAAAyC4gsAAAAAAIBBUHwB\nAAAAAAAwCIovAAAAAAAABkHxBQAAAAAAwCAovgAAAAAAABiE/wOqrC2zuOxtuAAAAABJRU5ErkJg\ngg==\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x7fa9a3c3cac8>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Create a distribution graph to identify the number of pure nuclear interacting or pure cytosol interacting RNAs  \n",
    "distrib_nuclearness = sorted([val[\"nucleus\"]-val[\"cytosol\"] for val in gene_loc_dict.values()])\n",
    "p= pl.hist(distrib_nuclearness)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
