{
 "metadata": {
  "name": "VCF analysis"
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "# Analysis of VCF file containing variants along rAAV genome from datasets generated by SSV-Seq"
     ]
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Create classes to handle Sample during VCF parsing"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "class Sample:\n",
      "\n",
      "    def __init__ (self, id, min_var_freq, ref_seq):\n",
      "        self.id = id\n",
      "        self.min_var_freq = min_var_freq\n",
      "        self.position_list = []\n",
      "        \n",
      "        # iterate other bases of the reference sequence\n",
      "        for base in ref_seq:\n",
      "            \n",
      "            # Create a dict for all 4 DNA base where the reference base is identified        \n",
      "            self.position_list.append( {\n",
      "                \"A\": {\"ref_base\": True, \"freq\" : 1} if base == \"A\" else {\"ref_base\": False, \"freq\" : 0},\n",
      "                \"C\": {\"ref_base\": True, \"freq\" : 1} if base == \"C\" else {\"ref_base\": False, \"freq\" : 0},\n",
      "                \"G\": {\"ref_base\": True, \"freq\" : 1} if base == \"G\" else {\"ref_base\": False, \"freq\" : 0},\n",
      "                \"T\": {\"ref_base\": True, \"freq\" : 1} if base == \"T\" else {\"ref_base\": False, \"freq\" : 0}})\n",
      "            \n",
      "    def add_variant (self, pos, base, freq):\n",
      "        if freq > self.min_var_freq:   \n",
      "            self.position_list[pos-1][base][\"freq\"] = freq"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 1
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Test Sample and Position Classes"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "a = Sample(\"test\", 0.01, \"ATATCGATC\")\n",
      "a.add_variant(3, 'T', 0.1)\n",
      "a.add_variant(3, 'C', 0.1)\n",
      "a.add_variant(3, 'A', 0.8)\n",
      "a.__dict__"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "pyout",
       "prompt_number": 2,
       "text": [
        "{'id': 'test',\n",
        " 'min_var_freq': 0.01,\n",
        " 'position_list': [{'A': {'freq': 1, 'ref_base': True},\n",
        "   'C': {'freq': 0, 'ref_base': False},\n",
        "   'G': {'freq': 0, 'ref_base': False},\n",
        "   'T': {'freq': 0, 'ref_base': False}},\n",
        "  {'A': {'freq': 0, 'ref_base': False},\n",
        "   'C': {'freq': 0, 'ref_base': False},\n",
        "   'G': {'freq': 0, 'ref_base': False},\n",
        "   'T': {'freq': 1, 'ref_base': True}},\n",
        "  {'A': {'freq': 0.8, 'ref_base': True},\n",
        "   'C': {'freq': 0.1, 'ref_base': False},\n",
        "   'G': {'freq': 0, 'ref_base': False},\n",
        "   'T': {'freq': 0.1, 'ref_base': False}},\n",
        "  {'A': {'freq': 0, 'ref_base': False},\n",
        "   'C': {'freq': 0, 'ref_base': False},\n",
        "   'G': {'freq': 0, 'ref_base': False},\n",
        "   'T': {'freq': 1, 'ref_base': True}},\n",
        "  {'A': {'freq': 0, 'ref_base': False},\n",
        "   'C': {'freq': 1, 'ref_base': True},\n",
        "   'G': {'freq': 0, 'ref_base': False},\n",
        "   'T': {'freq': 0, 'ref_base': False}},\n",
        "  {'A': {'freq': 0, 'ref_base': False},\n",
        "   'C': {'freq': 0, 'ref_base': False},\n",
        "   'G': {'freq': 1, 'ref_base': True},\n",
        "   'T': {'freq': 0, 'ref_base': False}},\n",
        "  {'A': {'freq': 1, 'ref_base': True},\n",
        "   'C': {'freq': 0, 'ref_base': False},\n",
        "   'G': {'freq': 0, 'ref_base': False},\n",
        "   'T': {'freq': 0, 'ref_base': False}},\n",
        "  {'A': {'freq': 0, 'ref_base': False},\n",
        "   'C': {'freq': 0, 'ref_base': False},\n",
        "   'G': {'freq': 0, 'ref_base': False},\n",
        "   'T': {'freq': 1, 'ref_base': True}},\n",
        "  {'A': {'freq': 0, 'ref_base': False},\n",
        "   'C': {'freq': 1, 'ref_base': True},\n",
        "   'G': {'freq': 0, 'ref_base': False},\n",
        "   'T': {'freq': 0, 'ref_base': False}}]}"
       ]
      }
     ],
     "prompt_number": 2
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Parse the reference sequence"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "from Bio import SeqIO\n",
      "ref_seq = str(SeqIO.read(open(\"./Cassette-AAV-CMV-GFP-hTK.fa\"), \"fasta\").seq)\n",
      "ref_seq"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "pyout",
       "prompt_number": 3,
       "text": [
        "'CTGCGCGCTCGCTCGCTCACTGAGGCCGCCCGGGCAAAGCCCGGGCGTCGGGCGACCTTTGGTCGCCCGGCCTCAGTGAGCGAGCGAGCGCGCAGAGAGGGAGTGGCCAACTCCATCACTAGGGGTTCCTTGTAGTTAATGATTAACCCGCCATGCTACTTATCTACGTAGCCATGCTCTAGGAAGATCTCGACGCGTTGACATTGATTATTGACTAGTTATTAATAGTAATCAATTACGGGGTCATTAGTTCATAGCCCATATATGGAGTTCCGCGTTACATAACTTACGGTAAATGGCCCGCCTGGCTGACCGCCCAACGACCCCCGCCCATTGACGTCAATAATGACGTATGTTCCCATAGTAACGCCAATAGGGACTTTCCATTGACGTCAATGGGTGGAGTATTTACGGTAAACTGCCCACTTGGCAGTACATCAAGTGTATCATATGCCAAGTACGCCCCCTATTGACGTCAATGACGGTAAATGGCCCGCCTGGCATTATGCCCAGTACATGACCTTATGGGACTTTCCTACTTGGCAGTACATCTACGTATTAGTCATCGCTATTACCATGGTGATGCGGTTTTGGCAGTACATCAATGGGCGTGGATAGCGGTTTGACTCACGGGGATTTCCAAGTCTCCACCCCATTGACGTCAATGGGAGTTTGTTTTGGCACCAAAATCAACGGGACTTTCCAAAATGTCGTAACAACTCCGCCCCATTGACGCAAATGGGCGGTAGGCGTGTACGGTGGGAGGTCTATATAAGCAGAGCTCTCTGGCTAACTAGAGAACCCACTGCTTACTGGCTTATCGAAATTAATACGACTCACTATAGGGAGACCCAAGCTGGCTAGCGACCATGGTGAGCAAGGGCGAGGAGCTGTTCACCGGGGTGGTGCCCATCCTGGTCGAGCTGGACGGCGACGTAAACGGCCACAAGTTCAGCGTGTCCGGCGAGGGCGAGGGCGATGCCACCTACGGCAAGCTGACCCTGAAGTTCATCTGCACCACCGGCAAGCTGCCCGTGCCCTGGCCCACCCTCGTGACCACCCTGACCTACGGCGTGCAGTGCTTCAGCCGCTACCCCGACCACATGAAGCAGCACGACTTCTTCAAGTCCGCCATGCCCGAAGGCTACGTCCAGGAGCGCACCATCTTCTTCAAGGACGACGGCAACTACAAGACCCGCGCCGAGGTGAAGTTCGAGGGCGACACCCTGGTGAACCGCATCGAGCTGAAGGGCATCGACTTCAAGGAGGACGGCAACATCCTGGGGCACAAGCTGGAGTACAACTACAACAGCCACAACGTCTATATCATGGCCGACAAGCAGAAGAACGGCATCAAGGTGAACTTCAAGATCCGCCACAACATCGAGGACGGCAGCGTGCAGCTCGCCGACCACTACCAGCAGAACACCCCCATCGGCGACGGCCCCGTGCTGCTGCCCGACAACCACTACCTGAGCACCCAGTCCGCCCTGAGCAAAGACCCCAACGAGAAGCGCGATCACATGGTCCTGCTGGAGTTCGTGACCGCCGCCGGGATCACTCTCGGCATGGACGAGCTGTACAAGTAAGTCGACCCCGCGAAATTAATACGACTCACTATAGGGAGACCACAACGGTTTCCCTCTAGCGGGATCAATTCCGCCCCCCCCCTAACGTTACTGGCCGAAGCCGCTTGGAATAAGGCCGGTGTGCGTTTGTCTATATGTTATTTTCCACCATATTGCCGTCTTTTGGCAATGTGAGGGCCCGGAAACCTGGCCCTGTCTTCTTGACGAGCATTCCTAGGGGTCTTTCCCCTCTCGCCAAAGGAATGCAAGGTCTGTTGAATGTCGTGAAGGAAGCAGTTCCTCTGGAAGCTTCTTGAAGACAAACAACGTCTGTAGCGACCCTTTGCAGGCAGCGGAACCCCCCACCTGGCGACAGGTGCCTCTGCGGCCAAAAGCCACGTGTATAAGATACACCTGCAAAGGCGGCACAACCCCAGTGCCACGTTGTGAGTTGGATAGTTGTGGAAAGAGTCAAATGGCTCTCCTCAAGCGTATTCAACAAGGGGCTGAAGGATGCCCAGAAGGTACCCCATTGTATGGGATCTGATCTGGGGCCTCGGTGCACATGCTTTACATGTGTTTAGTCGAGGTTAAAAAAACGTCTAGGCCCCCCGAACCACGGGGACGTGGTTTTCCTTTGAAAAACACGATGATATCATGAAAAAGCCTGAACTCACCGCGACGTCTGTCGAGAAGTTTCTGATCGAAAAGTTCGACAGCGTCTCCGACCTGATGCAGCTCTCGGAGGGCGAAGAATCTCGTGCTTTCAGCTTCGATGTAGGAGGGCGTGGATATGTCCTGCGGGTAAATAGCTGCGCCGATGGTTTCTACAAAGATCGTTATGTTTATCGGCACTTTGCATCGGCCGCGCTCCCGATTCCGGAAGTGCTTGACATTGGGGAATTCAGCGAGAGCCTGACCTATTGCATCTCCCGCCGTGCACGGGGTGTCACGTTGCAAGACCTGCCTGAAACCGAACTGCCCGCTGTTCTGCAGCCGGTCGCGGAGGCCATGGATGCGATCGCTGCGGCCGATCTTAGCCAGACGAGCGGGTTCGGCCCATTCGGACCGCAAGGAATCGGTCAATACACTACATGGCGTGATTTCATATGCGCGATTGCTGATCCCCATGTGTATCACTGGCAAACTGTGATGGACGACACCGTCAGTGCGTCCGTCGCGCAGGCTCTCGATGAGCTGATGCTTTGGGCCGAGGACTGCCCCGAAGTCCGGCACCTCGTGCACGCGGATTTCGGCTCCAACAATGTCCTGACGGACAATGGCCGCATAACAGCGGTCATTGACTGGAGCGAGGCGATGTTCGGGGATTCCCAATACGAGGTCGCCAACATCTTCTTCTGGAGGCCGTGGTTGGCTTGTATGGAGCAGCAGACGCGCTACTTCGAGCGGAGGCATCCGGAGCTTGCAGGATCGCCGCGGCTCCGGGCGTATATGCTCCGCATTGGTCTTGACCAACTCTATCAGAGCTTGGTTGACGGCAATTTCGATGATGCAGCTTGGGCGCAGGGTCGATGCGACGCAATCGTCCGATCCGGAGCCGGGACTGTCGGGCGTACACAAATCGCCCGCAGAAGCGCGGCCGTCTGGACCGATGGCTGTGTAGAAGTCGCGTCTGCGTTCGACCAGGCTGCGCGTTCTCGCGGCCATAGCAACCGACGTACGGCGTTGCGCCCTCGCCGGCAGCAAGAAGCCACGGAAGTCCGCCCGGAGCAGAAAATGCCCACGCTACTGCGGGTTTATATAGACGGTCCCCACGGGATGGGGAAAACCACCACCACGCAACTGCTGGTGGCCCTGGGTTCGCGCGACGATATCGTCTACGTACCCGAGCCGATGACTTACTGGCGGGTGCTGGGGGCTTCCGAGACAATCGCGAACATCTACACCACACAACACCGCCTCGACCAGGGTGAGATATCGGCCGGGGACGCGGCGGTGGTAATGACAAGCGCCCAGATAACAATGGGCATGCCTTATGCCGTGACCGACGCCGTTCTGGCTCCTCATATCGGGGGGGAGGCTGGGAGCTCACATGCCCCGCCCCCGGCCCTCACCCTCATCTTCGACCGCCATCCCATCGCCGCCCTCCTGTGCTACCCGGCCGCGCGGTACCTTATGGGCAGCATGACCCCCCAGGCCGTGCTGGCGTTCGTGGCCCTCATCCCGCCGACCTTGCCCGGCACCAACATCGTGCTTGGGGCCCTTCCGGAGGACAGACACATCGACCGCCTGGCCAAACGCCAGCGCCCCGGCGAGCGGCTGGACCTGGCTATGCTGGCTGCGATTCGCCGCGTTTACGGGCTACTTGCCAATACGGTGCGGTATCTGCAGTGCGGCGGGTCGTGGCGGGAGGACTGGGGACAGCTTTCGGGGACGGCCGTGCCGCCCCAGGGTGCCGAGCCCCAGAGCAACGCGGGCCCACGACCCCATATCGGGGACACGTTATTTACCCTGTTTCGGGCCCCCGAGTTGCTGGCCCCCAACGGCGACCTGTATAACGTGTTTGCCTGGGCCTTGGACGTCTTGGCCAAACGCCTCCGTTCCATGCACGTCTTTATCCTGGATTACGACCAATCGCCCGCCGGCTGCCGGGACGCCCTGCTGCAACTTACCTCCGGGATGGTCCAGACCCACGTCACCACCCCCGGCTCCATACCGACGATATGCGACCTGGCGCGCACGTTTGCCCGGGAGATGGGGGAGGCTAACTGAAAATCGATGGATCCACTAGTTCTAGAGGGCCCTATTCTATAGTGTCACCTAAATGCTAGAGCTCGCTGATCAGCCTCGACTGTGCCTTCTAGTTGCCAGCCATCTGTTGTTTGCCCCTCCCCCGTGCCTTCCTTGACCCTGGAAGGTGCCACTCCCACTGTCCTTTCCTAATAAAATGAGGAAATTGCATCGCATTGTCTGAGTAGGTGTCATTCTATTCTGGGGGGTGGGGTGGGGCAGGACAGCAAGGGGGAGGATTGGGAAGACAATAGCAGGCATGCTGGGGATGCGGTGGGCTCTATGGCTTCTGAGGCGGAAAGAACCAGGTAGATAAGTAGCATGGCGGGTTAATCATTAACTACAAGGAACCCCTAGTGATGGAGTTGGCCACTCCCTCTCTGCGCGCTCGCTCGCTCACTGAGGCCGGGCGACCAAAGGTCGCCCGACGCCCGGGCTTTGCCCGGGCGGCCTCAGTGAGCGAGCGAGCGCGCAG'"
       ]
      }
     ],
     "prompt_number": 3
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Define the min freq"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "min_freq = float(1)/1000\n",
      "print min_freq"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "0.001\n"
       ]
      }
     ],
     "prompt_number": 4
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Define the Sample in VCF file"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "sample_dict = {\n",
      " 'C1_AAV': Sample('C1_AAV', min_freq, ref_seq),\n",
      " 'RUN1_S7_AAV': Sample('RUN1_S7_AAV', min_freq, ref_seq),\n",
      " 'RUN2_S8_AAV': Sample('RUN2_S8_AAV', min_freq, ref_seq),\n",
      " 'RUN1_S2_AAV': Sample('RUN1_S2_AAV', min_freq, ref_seq),\n",
      " 'RUN2_S1_AAV': Sample('RUN2_S1_AAV', min_freq, ref_seq),\n",
      " 'RUN1_S4_AAV': Sample('RUN1_S4_AAV', min_freq, ref_seq),\n",
      " 'RUN2_S2_AAV': Sample('RUN2_S2_AAV', min_freq, ref_seq),\n",
      " 'RUN1_S6_AAV': Sample('RUN1_S6_AAV', min_freq, ref_seq),\n",
      " 'RUN2_S3_AAV': Sample('RUN2_S3_AAV', min_freq, ref_seq),\n",
      " 'RUN1_S8_AAV': Sample('RUN1_S8_AAV', min_freq, ref_seq),\n",
      " 'RUN2_S4_AAV': Sample('RUN2_S4_AAV', min_freq, ref_seq),\n",
      " 'RUN1_S3_AAV': Sample('RUN1_S3_AAV', min_freq, ref_seq),\n",
      " 'RUN2_S5_AAV': Sample('RUN2_S5_AAV', min_freq, ref_seq),\n",
      " 'RUN1_S5_AAV': Sample('RUN1_S5_AAV', min_freq, ref_seq),\n",
      " 'RUN2_S6_AAV': Sample('RUN2_S6_AAV', min_freq, ref_seq)}\n",
      " \n",
      "# 'RUN2_S7_AAV': \"Negative Control 2\", # Negative control have not interest for this analysis\n",
      "# 'RUN1_S1_AAV': \"Negative Control 1\", # Negative control have not interest for this analysis\n",
      "# 'RUN2_S9_AAV': Sample('', \"Internal Normalizer 3\", min_freq, ref_seq),"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 5
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Open vcf file and parse general metadata"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import HTSeq\n",
      "vcfr = HTSeq.VCF_Reader( \"./NoIndel.vcf\" ) \n",
      "vcfr.parse_meta() \n",
      "vcfr.make_info_dict() "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 6
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "Parse VCF variants and add to Sample objects if an alternative base if found"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for variant in vcfr:\n",
      "    # Extract information for samples in sample dict\n",
      "    for name, sample in sample_dict.items():\n",
      "        \n",
      "        # Extract genotypes from the INFO field for the current sample\n",
      "        GT = [int(i) for i in variant.samples[name]['GT'].split('/')]\n",
      "        #print variant.samples[name]\n",
      "        # Extract more informations only if needed\n",
      "        if len(GT) > 1:\n",
      "            DPG = [int(i) for i in variant.samples[name]['DPG'].split(',')]\n",
      "            DP = int(variant.samples[name]['DP'])\n",
      "            base = [variant.ref]+variant.alt\n",
      "            \n",
      "            # Modifify sample for each alternative genotype\n",
      "            for i in range (len (GT)):\n",
      "                sample.add_variant(\n",
      "                    pos = variant.pos.pos,\n",
      "                    base = base[GT[i]],\n",
      "                    freq = float(DPG[i])/DP)\n",
      "    "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 7
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "sample_group = {\n",
      "    \"Internal Normalizer\" : ['RUN1_S7_AAV', 'RUN2_S8_AAV'],}\n",
      "    "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 8
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "sample_group = {\n",
      "    \"IEX\" : ['RUN1_S3_AAV', 'RUN2_S5_AAV', 'RUN1_S5_AAV', 'RUN2_S6_AAV']}"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 9
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "sample_group = {\n",
      "    \"Internal Normalizer\" : ['RUN1_S7_AAV', 'RUN2_S8_AAV'],\n",
      "    \"CsCl-\" : ['RUN1_S2_AAV', 'RUN2_S1_AAV'],\n",
      "    \"CsCl+\" : ['RUN1_S4_AAV', 'RUN2_S2_AAV'],\n",
      "    \"AVB-\" : ['RUN1_S6_AAV', 'RUN2_S3_AAV'],\n",
      "    \"AVB+\" : ['RUN1_S8_AAV', 'RUN2_S4_AAV'],\n",
      "    \"IEX-\" : ['RUN1_S3_AAV', 'RUN2_S5_AAV'],\n",
      "    \"IEX+\" : ['RUN1_S5_AAV', 'RUN2_S6_AAV'],\n",
      "    \"CsCl\" : ['RUN1_S2_AAV', 'RUN2_S1_AAV', 'RUN1_S4_AAV', 'RUN2_S2_AAV'],\n",
      "    \"AVB\" : ['RUN1_S6_AAV', 'RUN2_S3_AAV', 'RUN1_S8_AAV', 'RUN2_S4_AAV'],\n",
      "    \"IEX\" : ['RUN1_S3_AAV', 'RUN2_S5_AAV', 'RUN1_S5_AAV', 'RUN2_S6_AAV'],\n",
      "    \"ALL_AAV\" : ['RUN1_S2_AAV', 'RUN2_S1_AAV', 'RUN1_S4_AAV', 'RUN2_S2_AAV','RUN1_S6_AAV', 'RUN2_S3_AAV', 'RUN1_S8_AAV', 'RUN2_S4_AAV', 'RUN1_S3_AAV', 'RUN2_S5_AAV', 'RUN1_S5_AAV', 'RUN2_S6_AAV']}"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 13
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for name, group in sample_group.items():\n",
      "    sample = Sample('name', min_freq, ref_seq)\n",
      "    \n",
      "    # For each position along the length of reference the sequence\n",
      "    for i in range (len(ref_seq)):\n",
      "        for base in [\"A\", \"C\", \"G\", \"T\"]:\n",
      "                        \n",
      "            # Not interested by the frequency of the reference base\n",
      "            if sample.position_list[i][base]['ref_base']:\n",
      "                sample.position_list[i][base][\"freq\"] = 0.0\n",
      "                \n",
      "            # Extract the list of frequency for this base in all samples\n",
      "            else: \n",
      "                var_freq_list = [sample_dict[var_name].position_list[i][base][\"freq\"] for var_name in group]\n",
      "                #print (var_freq_list)\n",
      "                \n",
      "                # At least half of the samples in the list must have the variant with a frequency > 0 else it is considered null\n",
      "                if len([j for j in var_freq_list if j != 0]) >= len(var_freq_list)/2:\n",
      "                    sample.position_list[i][base][\"freq\"] = sum(var_freq_list)/len(var_freq_list)\n",
      "                    #print (\"TRUE\")\n",
      "                else:\n",
      "                    sample.position_list[i][base][\"freq\"] = 0.0\n",
      "                \n",
      "    # Write 1 report per sample\n",
      "    with open (name+\"_report.csv\", \"wb\") as report:\n",
      "        report.write(\"Pos\\tAlt_A\\tAlt_C\\tAlt_G\\tAlt_T\\tsum\\n\")\n",
      "        for n, p in enumerate(sample.position_list):\n",
      "            report.write (\"{}\\t{}\\t{}\\t{}\\t{}\\t{}\\n\".format(\n",
      "                n, p[\"A\"]['freq'], p[\"C\"]['freq'], p[\"G\"]['freq'], p[\"T\"]['freq'],\n",
      "                p[\"A\"]['freq'] + p[\"C\"]['freq'] + p[\"G\"]['freq'] + p[\"T\"]['freq']))\n",
      "        \n",
      "    "
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 14
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Plot 1 graphs per sample\n",
      "    from matplotlib import pyplot as plt\n",
      "    \n",
      "    fig = plt.figure(figsize=(500, 50), dpi=100)\n",
      "    \n",
      "    A_list = [int(p[\"A\"]['freq']*1000) for p in sample.position_list]\n",
      "    print A_list\n",
      "    C_list = [p[\"C\"]['freq'] for p in sample.position_list]\n",
      "    G_list = [p[\"G\"]['freq'] for p in sample.position_list]\n",
      "    T_list = [p[\"T\"]['freq'] for p in sample.position_list]\n",
      "    \n",
      "    plt.bar(range(len(A_list)), A_list, color='r')#, alpha=0.5, log=True)\n",
      "    plt.bar(range(len(C_list)), C_list, color='g')#, alpha=0.5, bottom=A_bar)#, log=True)\n",
      "    plt.bar(range(len(G_list)), G_list, color='b')#, alpha=0.5, bottom=C_bar)#, log=True)\n",
      "    plt.bar(range(len(T_list)), T_list, color='o')#, alpha=0.5, bottom=G_bar)#, log=True)\n",
      "    \n",
      "    plt.ylabel('Frequency')\n",
      "    plt.title('Frequency of alternative base ')\n",
      "    fig.savefig(\"test.svg\", format = svg)\n",
      "    \n",
      "    #plt.legend( (A_bar[0], C_bar[0], G_bar[0] , T_bar[0]), ('A','C','G','T') )\n",
      "    #plt.show()\n",
      "    "
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "from matplotlib import pyplot as plt\n",
      "\n",
      "s1=[0,1,2,3,4,5,6,7,8,9,0,1,2,3,4,5,6,7,8,9,0,1,2,3,4,5,6,7,8,9,0,1,2,3,4,5,6,7,8,9,0,1,2,3,4,5,6,7,8,9,0,1,2,3,4,5,6,7,8,9,0,1,2,3,4,5,6,7,8,9,0,1,2,3,4,5,6,7,8,9,0,1,2,3,4,5,6,7,8,9,0,1,2,3,4,5,6,7,8,9,]\n",
      "s2=[3,4,5,6,7,8,9,10,11,1,3,4,5,6,7,8,9,10,11,1,3,4,5,6,7,8,9,10,11,1,3,4,5,6,7,8,9,10,11,1,3,4,5,6,7,8,9,10,11,1,3,4,5,6,7,8,9,10,11,1,3,4,5,6,7,8,9,10,11,1,3,4,5,6,7,8,9,10,11,1,3,4,5,6,7,8,9,10,11,1,3,4,5,6,7,8,9,10,11,1,]\n",
      "\n",
      "plt.bar(range(100), s1, color='r')\n",
      "plt.bar(range(100), s2, color='g', bottom=s1)\n",
      "plt.show()"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 1
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}